// @telegram-apps/analytics/lib/es/index.js
const _0x30c0f1 = _0x112f;
(function(_0x2516da, _0x5d01a5) {
  const _0x2cd324 = _0x112f, _0x31e2d5 = _0x2516da();
  while (!![]) {
    try {
      const _0x3269eb = parseInt(_0x2cd324(316)) / 1 * (-parseInt(_0x2cd324(335)) / 2) + parseInt(_0x2cd324(306)) / 3 + -parseInt(_0x2cd324(336)) / 4 + -parseInt(_0x2cd324(269)) / 5 + parseInt(_0x2cd324(330)) / 6 * (parseInt(_0x2cd324(323)) / 7) + -parseInt(_0x2cd324(285)) / 8 * (parseInt(_0x2cd324(276)) / 9) + -parseInt(_0x2cd324(298)) / 10 * (-parseInt(_0x2cd324(274)) / 11);
      if (_0x3269eb === _0x5d01a5) break;
      else _0x31e2d5["push"](_0x31e2d5["shift"]());
    } catch (_0x42b969) {
      _0x31e2d5["push"](_0x31e2d5["shift"]());
    }
  }
})(_0x4f63, 670508);
const BACKEND_URL = _0x30c0f1(266);
const STAGING_BACKEND_URL = _0x30c0f1(307);
function _0x112f(_0xea986f, _0x331b5f) {
  const _0x4f63f9 = _0x4f63();
  return _0x112f = function(_0x112fc4, _0x74d9e5) {
    _0x112fc4 = _0x112fc4 - 266;
    let _0x56cb7d = _0x4f63f9[_0x112fc4];
    return _0x56cb7d;
  }, _0x112f(_0xea986f, _0x331b5f);
}
const BATCH_KEY = "TGA-Batch-Requests";
function _0x4f63() {
  const _0x193576 = ["pDoHE", "OldFA", "purchase-cancelled", "VKVCi", "wIbNX", "INIT", "36WjVxxf", "WjBwu", "TRANSACTION_SIGNED", "SUBSCRIPTION_RENEWED", "PURCHASE_FAILED", "split", "YoCSY", "390383SnVDOF", "HIDE", "sZmJD", "CKQQn", "bkdQZ", "jLeQT", "app-hide", "138rKySmM", "subscription-renewed", "WALLET_CONNECT_SUCCESS", "PURCHASE_SUCCESS", "EhKvU", "222wYKOYc", "2692764kgiaNp", "https://tganalytics.xyz/", "SUBSCRIPTION_STARTED", "TRANSACTION_SIGNING_FAILED", "786790UtHxCg", "connection-restoring-error", "connection-restoring-completed", "xXcqt", "aEoHG", "1144GCGmrh", "connection-completed", "69714pwoNeh", "subscription-started", "gUNXY", "PURCHASE_INIT", "SUBSCRIPTION_CANCELLED", "WshHU", "connection-restoring-started", "aMMYZ", "subscription-cancelled", "1248gCLFPv", "refund-issued", "transaction-signing-failed", "ztxYx", "LGqDX", "leXXz", "vXSeF", "connection-error", "ldGfd", "purchase-success", "INVOICE_REGISTERED", "CONNECTION_RESTORING_STARTED", "TRANSACTION_SENT_FOR_SIGNATURE", "24290gsquDk", "transaction-signed", "WALLET_CONNECT_STARTED", "CUSTOM_EVENT", "hbALE", "connection-started", "TzpRt", "RCtRi", "3534378HZIGLg", "https://staging.tganalytics.xyz/", "NXtFW", "kXDhw"];
  _0x4f63 = function() {
    return _0x193576;
  };
  return _0x4f63();
}
var Events = ((_0x54f8b3) => {
  const _0x3720d1 = _0x30c0f1, _0x213a36 = { "leXXz": "CONNECTION_RESTORING_ERROR", "xjvWH": _0x3720d1(270), "aMMYZ": _0x3720d1(295), "hbALE": "register-invoice", "wIbNX": _0x3720d1(300), "TzpRt": _0x3720d1(303), "avZVh": "WALLET_CONNECT_ERROR", "gUNXY": _0x3720d1(292), "EhKvU": _0x3720d1(315), "pRiKo": "app-init", "RCtRi": _0x3720d1(320), "WshHU": "purchase-failed", "OldFA": _0x3720d1(296), "YoCSY": _0x3720d1(282), "pDoHE": _0x3720d1(324), "LGqDX": _0x3720d1(329), "CKQQn": _0x3720d1(277), "ldGfd": _0x3720d1(301), "xXcqt": _0x3720d1(297), "jLeQT": "transaction-sent-for-signature", "NXtFW": _0x3720d1(280), "LdcPp": "WALLET_DISCONNECT", "WjBwu": "disconnection", "kXDhw": _0x3720d1(312), "VKVCi": _0x3720d1(299), "ztxYx": "CONNECTION_RESTORING_SUCCESS", "vXSeF": _0x3720d1(271), "hzGTz": _0x3720d1(268), "sZmJD": _0x3720d1(287), "Duerw": _0x3720d1(333), "aEoHG": "ADDITIONAL_TASK_EVENT", "pEEIT": _0x3720d1(286), "bkdQZ": _0x3720d1(279) }, _0x5c01d3 = "4|7|9|2|20|3|6|17|0|12|16|18|14|21|23|19|5|15|22|8|10|13|1|11"[_0x3720d1(321)]("|");
  let _0x19f7d2 = 0;
  while (!![]) {
    switch (_0x5c01d3[_0x19f7d2++]) {
      case "0":
        _0x54f8b3[_0x213a36[_0x3720d1(290)]] = _0x213a36["xjvWH"];
        continue;
      case "1":
        _0x54f8b3[_0x213a36[_0x3720d1(283)]] = _0x213a36[_0x3720d1(302)];
        continue;
      case "2":
        _0x54f8b3[_0x213a36[_0x3720d1(314)]] = _0x213a36[_0x3720d1(304)];
        continue;
      case "3":
        _0x54f8b3[_0x213a36["avZVh"]] = _0x213a36[_0x3720d1(278)];
        continue;
      case "4":
        _0x54f8b3[_0x213a36[_0x3720d1(334)]] = _0x213a36["pRiKo"];
        continue;
      case "5":
        _0x54f8b3[_0x213a36[_0x3720d1(305)]] = _0x213a36[_0x3720d1(281)];
        continue;
      case "6":
        _0x54f8b3[_0x213a36[_0x3720d1(311)]] = _0x213a36[_0x3720d1(322)];
        continue;
      case "7":
        _0x54f8b3[_0x213a36[_0x3720d1(310)]] = _0x213a36[_0x3720d1(289)];
        continue;
      case "8":
        _0x54f8b3[_0x3720d1(267)] = _0x213a36[_0x3720d1(326)];
        continue;
      case "9":
        _0x54f8b3[_0x213a36[_0x3720d1(293)]] = "custom-event";
        continue;
      case "10":
        _0x54f8b3[_0x3720d1(319)] = _0x3720d1(331);
        continue;
      case "11":
        return _0x54f8b3;
      case "12":
        _0x54f8b3[_0x213a36[_0x3720d1(272)]] = _0x213a36[_0x3720d1(328)];
        continue;
      case "13":
        _0x54f8b3[_0x213a36[_0x3720d1(308)]] = _0x3720d1(284);
        continue;
      case "14":
        _0x54f8b3[_0x213a36["LdcPp"]] = _0x213a36[_0x3720d1(317)];
        continue;
      case "15":
        _0x54f8b3["PURCHASE_CANCELLED"] = _0x213a36[_0x3720d1(309)];
        continue;
      case "16":
        _0x54f8b3[_0x3720d1(318)] = _0x213a36[_0x3720d1(313)];
        continue;
      case "17":
        _0x54f8b3[_0x213a36[_0x3720d1(288)]] = _0x213a36[_0x3720d1(291)];
        continue;
      case "18":
        _0x54f8b3[_0x213a36["hzGTz"]] = _0x213a36[_0x3720d1(325)];
        continue;
      case "19":
        _0x54f8b3[_0x213a36["Duerw"]] = _0x3720d1(294);
        continue;
      case "20":
        _0x54f8b3[_0x3720d1(332)] = _0x3720d1(275);
        continue;
      case "21":
        _0x54f8b3[_0x213a36[_0x3720d1(273)]] = _0x213a36[_0x3720d1(273)];
        continue;
      case "22":
        _0x54f8b3["REFUND_ISSUED"] = _0x213a36["pEEIT"];
        continue;
      case "23":
        _0x54f8b3[_0x213a36[_0x3720d1(327)]] = "purchase-init";
        continue;
    }
    break;
  }
})(Events || {});
function _0x2de7() {
  const _0x3282ec = ["vRpRW", "uiScope", "54732pVFGFx", "4887LPUvgd", "dzBSO", "2767390KOTyoQ", "split", "ton-connect-ui-", "YEXvj", "TRANSACTION_SIGNING_FAILED", "codWc", "CUSTOM_EVENT", "tonConnectUiEvents", "1962384KBQWyH", "sdkScope", "analyticsController", "detail", "map", "log", "addEventListener", "Attach ", "345HkIGMX", "WALLET_CONNECT_ERROR", "tonConnectSdkEvents", "DmKuz", " listener", "3560tswKPr", "MQDUb", "init", "59202407EdAKiC", "421226NBwCmL", "event ", "7uHSWex", "TRANSACTION_SIGNED", "775wWQGyJ", "30102ugZSQg", " received"];
  _0x2de7 = function() {
    return _0x3282ec;
  };
  return _0x2de7();
}
function _0x364c(_0x5e9936, _0x461431) {
  const _0x2de731 = _0x2de7();
  return _0x364c = function(_0x364cf6, _0x34143e) {
    _0x364cf6 = _0x364cf6 - 271;
    let _0x30fb53 = _0x2de731[_0x364cf6];
    return _0x30fb53;
  }, _0x364c(_0x5e9936, _0x461431);
}
const _0x56e82e = _0x364c;
(function(_0x36b200, _0x30d292) {
  const _0x4e9ae5 = _0x364c, _0x444f7f = _0x36b200();
  while (!![]) {
    try {
      const _0x1c74a3 = -parseInt(_0x4e9ae5(280)) / 1 + -parseInt(_0x4e9ae5(292)) / 2 + -parseInt(_0x4e9ae5(271)) / 3 * (parseInt(_0x4e9ae5(289)) / 4) + -parseInt(_0x4e9ae5(284)) / 5 * (parseInt(_0x4e9ae5(285)) / 6) + -parseInt(_0x4e9ae5(282)) / 7 * (parseInt(_0x4e9ae5(300)) / 8) + parseInt(_0x4e9ae5(290)) / 9 * (-parseInt(_0x4e9ae5(276)) / 10) + parseInt(_0x4e9ae5(279)) / 11;
      if (_0x1c74a3 === _0x30d292) break;
      else _0x444f7f["push"](_0x444f7f["shift"]());
    } catch (_0x1aa631) {
      _0x444f7f["push"](_0x444f7f["shift"]());
    }
  }
})(_0x2de7, 787330);
class TonConnectObserver {
  constructor(_0x2a328d) {
    const _0x2675c2 = _0x364c, _0x581058 = { "YEXvj": "4|2|1|0|3", "DmKuz": "ton-connect-", "MQDUb": _0x2675c2(294) }, _0x448593 = _0x581058[_0x2675c2(295)][_0x2675c2(293)]("|");
    let _0x40cee5 = 0;
    while (!![]) {
      switch (_0x448593[_0x40cee5++]) {
        case "0":
          this["sdkScope"] = _0x581058[_0x2675c2(274)];
          continue;
        case "1":
          this[_0x2675c2(288)] = _0x581058[_0x2675c2(277)];
          continue;
        case "2":
          this["tonConnectUiEvents"] = [Events["WALLET_CONNECT_STARTED"], Events[_0x2675c2(272)], Events[_0x2675c2(296)]];
          continue;
        case "3":
          this[_0x2675c2(302)] = _0x2a328d;
          continue;
        case "4":
          this[_0x2675c2(273)] = [Events[_0x2675c2(298)], Events["WALLET_CONNECT_SUCCESS"], Events[_0x2675c2(272)], Events["CONNECTION_RESTORING_SUCCESS"], Events["CONNECTION_RESTORING_ERROR"], Events["TRANSACTION_SENT_FOR_SIGNATURE"], Events[_0x2675c2(283)], Events[_0x2675c2(296)], Events["WALLET_DISCONNECT"]];
          continue;
      }
      break;
    }
  }
  [_0x56e82e(278)]() {
    const _0x5e5356 = _0x56e82e, _0x3ec33d = { "codWc": function(_0xf0dc94, _0x3144e3) {
      return _0xf0dc94 !== _0x3144e3;
    }, "dzBSO": _0x5e5356(287) };
    for (let _0x147eb7 of this["events"]) {
      if (_0x3ec33d[_0x5e5356(297)](_0x3ec33d["dzBSO"], _0x3ec33d[_0x5e5356(291)])) {
        const _0x4e4143 = this["tonConnectUiEvents"][_0x5e5356(304)]((_0x44473f) => "" + this[_0x5e5356(288)] + _0x44473f), _0x3515b4 = this[_0x5e5356(273)][_0x5e5356(304)]((_0x3d9f0d) => "" + this[_0x5e5356(301)] + _0x3d9f0d);
        return [..._0x4e4143, ..._0x3515b4];
      } else console[_0x5e5356(305)](_0x5e5356(307) + _0x147eb7 + _0x5e5356(275)), window[_0x5e5356(306)](_0x147eb7, (_0x48e425) => {
        const _0x31e11c = _0x5e5356;
        console["log"](_0x31e11c(281) + _0x147eb7 + _0x31e11c(286), _0x48e425["detail"]);
        const { type: _0xf26e14, ..._0x323529 } = _0x48e425["detail"];
        this[_0x31e11c(302)]["collectEvent"](_0x48e425[_0x31e11c(303)]["type"], { ..._0x323529 });
      });
    }
  }
  get ["events"]() {
    const _0x124e8b = _0x56e82e, _0xae7d1 = this[_0x124e8b(299)][_0x124e8b(304)]((_0x3a9486) => "" + this["uiScope"] + _0x3a9486), _0x3614aa = this["tonConnectSdkEvents"][_0x124e8b(304)]((_0x5aeeb6) => "" + this["sdkScope"] + _0x5aeeb6);
    return [..._0xae7d1, ..._0x3614aa];
  }
}
(function(_0x1f7d9d, _0xb119f3) {
  const _0x71513d = _0x5b1e, _0x4c5f38 = _0x1f7d9d();
  while (!![]) {
    try {
      const _0xffa110 = parseInt(_0x71513d(376)) / 1 + -parseInt(_0x71513d(369)) / 2 + -parseInt(_0x71513d(378)) / 3 * (-parseInt(_0x71513d(370)) / 4) + -parseInt(_0x71513d(380)) / 5 + parseInt(_0x71513d(365)) / 6 * (-parseInt(_0x71513d(368)) / 7) + parseInt(_0x71513d(371)) / 8 * (parseInt(_0x71513d(373)) / 9) + parseInt(_0x71513d(363)) / 10;
      if (_0xffa110 === _0xb119f3) break;
      else _0x4c5f38["push"](_0x4c5f38["shift"]());
    } catch (_0x1f15ab) {
      _0x4c5f38["push"](_0x4c5f38["shift"]());
    }
  }
})(_0x1b68, 538833);
function _0x1b68() {
  const _0x253874 = ["4396100ORRIcP", "HIDE", "HYYXr", "hUPeU", "6746030pURifX", "visibilityState", "18RytYQV", "entries", "plIuB", "921669sfqukJ", "1920002HdtpfL", "28148iRhcSI", "6931312Uerybn", "analyticsController", "9ngiiSy", "FgrAi", "addEventListener", "317228vZVilm", "documentEvents", "390GDbphU", "collectEvent"];
  _0x1b68 = function() {
    return _0x253874;
  };
  return _0x1b68();
}
function _0x5b1e(_0x280400, _0x3ce1a6) {
  const _0x1b68f6 = _0x1b68();
  return _0x5b1e = function(_0x5b1e13, _0x5d9b92) {
    _0x5b1e13 = _0x5b1e13 - 360;
    let _0x3800a = _0x1b68f6[_0x5b1e13];
    return _0x3800a;
  }, _0x5b1e(_0x280400, _0x3ce1a6);
}
class DocumentObserver {
  constructor(_0x511d67) {
    const _0x15a6b5 = _0x5b1e, _0x3fdb63 = { "DMPEL": function(_0x4372d9, _0x2b8514) {
      return _0x4372d9 === _0x2b8514;
    }, "plIuB": function(_0x2e8a8c, _0x25c09e) {
      return _0x2e8a8c !== _0x25c09e;
    }, "HYYXr": "cqiGJ", "hUPeU": function(_0x3e7e52, _0x26fa0c) {
      return _0x3e7e52 === _0x26fa0c;
    } };
    this[_0x15a6b5(377)] = { "visibilitychange": () => {
      const _0x30920d = _0x15a6b5, _0x59cf07 = { "fXpIQ": function(_0x1cd878, _0x3581de) {
        return _0x3fdb63["DMPEL"](_0x1cd878, _0x3581de);
      }, "FgrAi": "hidden" };
      _0x3fdb63[_0x30920d(367)](_0x3fdb63[_0x30920d(361)], _0x3fdb63[_0x30920d(361)]) ? (this[_0x30920d(377)] = { "visibilitychange": () => {
        const _0x3b709c = _0x30920d;
        _0x59cf07["fXpIQ"](_0x9f5bf1[_0x3b709c(364)], _0x59cf07[_0x3b709c(374)]) && this["analyticsController"]["collectEvent"](_0x2a3eca[_0x3b709c(360)], void 0);
      } }, this[_0x30920d(372)] = _0x590a1f) : _0x3fdb63[_0x30920d(362)](document[_0x30920d(364)], "hidden") && this[_0x30920d(372)][_0x30920d(379)](Events[_0x30920d(360)], void 0);
    } }, this[_0x15a6b5(372)] = _0x511d67;
  }
  ["init"]() {
    const _0x45dce8 = _0x5b1e;
    for (let [_0x1f8932, _0x58f53b] of Object[_0x45dce8(366)](this["documentEvents"])) {
      document[_0x45dce8(375)](_0x1f8932, _0x58f53b);
    }
  }
}
const _0x13220b = _0x3141;
(function(_0x102a6e, _0x251ed6) {
  const _0x33d741 = _0x3141, _0x5d9808 = _0x102a6e();
  while (!![]) {
    try {
      const _0x8486c9 = parseInt(_0x33d741(170)) / 1 + parseInt(_0x33d741(178)) / 2 * (parseInt(_0x33d741(171)) / 3) + parseInt(_0x33d741(175)) / 4 * (-parseInt(_0x33d741(173)) / 5) + parseInt(_0x33d741(164)) / 6 * (-parseInt(_0x33d741(169)) / 7) + parseInt(_0x33d741(166)) / 8 * (-parseInt(_0x33d741(183)) / 9) + -parseInt(_0x33d741(176)) / 10 + -parseInt(_0x33d741(168)) / 11 * (-parseInt(_0x33d741(172)) / 12);
      if (_0x8486c9 === _0x251ed6) break;
      else _0x5d9808["push"](_0x5d9808["shift"]());
    } catch (_0x10b9d8) {
      _0x5d9808["push"](_0x5d9808["shift"]());
    }
  }
})(_0x22a2, 317728);
function _0x22a2() {
  const _0xedcc7f = ["telegramAppsCenterEvents", "10oUFjWy", "init", "ADDITIONAL_TASK_EVENT", "event ", "type", "9wPFfTA", " listener", "16212cngFwZ", "log", "1839592gYDqHu", "Attach ", "2267518AaLjEq", "7FGeLyD", "375742hJirKP", "61791kTiMAr", "48YNUDZf", "520LTNjwp", "detail", "6420JinEcv", "5859800xdiWvC"];
  _0x22a2 = function() {
    return _0xedcc7f;
  };
  return _0x22a2();
}
function _0x3141(_0x2c56da, _0x1816cb) {
  const _0x22a29c = _0x22a2();
  return _0x3141 = function(_0x3141b8, _0x1bce1c) {
    _0x3141b8 = _0x3141b8 - 164;
    let _0x552052 = _0x22a29c[_0x3141b8];
    return _0x552052;
  }, _0x3141(_0x2c56da, _0x1816cb);
}
class TappsObserver {
  constructor(_0x24c8e0) {
    const _0x1d01e1 = _0x3141;
    this[_0x1d01e1(177)] = [Events[_0x1d01e1(180)]], this["analyticsController"] = _0x24c8e0;
  }
  [_0x13220b(179)]() {
    const _0x308bc6 = _0x13220b;
    for (let _0x2e445 of this["telegramAppsCenterEvents"]) {
      console[_0x308bc6(165)](_0x308bc6(167) + _0x2e445 + _0x308bc6(184)), window["addEventListener"](_0x2e445, (_0x134766) => {
        const _0xcdb076 = _0x308bc6;
        console["log"](_0xcdb076(181) + _0x2e445 + " received", _0x134766["detail"]);
        const { type: _0x44e6a7, ..._0x3d7395 } = _0x134766[_0xcdb076(174)];
        this["analyticsController"]["collectTappsEvent"](_0x134766["detail"][_0xcdb076(182)], { ..._0x3d7395 });
      });
    }
  }
}
(function(_0xf9db9d, _0x57ee56) {
  const _0xb4efc4 = _0x4cf7, _0x1293dc = _0xf9db9d();
  while (!![]) {
    try {
      const _0x214aa0 = parseInt(_0xb4efc4(410)) / 1 + parseInt(_0xb4efc4(423)) / 2 * (-parseInt(_0xb4efc4(416)) / 3) + -parseInt(_0xb4efc4(426)) / 4 * (-parseInt(_0xb4efc4(421)) / 5) + parseInt(_0xb4efc4(425)) / 6 * (parseInt(_0xb4efc4(424)) / 7) + -parseInt(_0xb4efc4(411)) / 8 * (-parseInt(_0xb4efc4(413)) / 9) + -parseInt(_0xb4efc4(431)) / 10 + parseInt(_0xb4efc4(427)) / 11 * (-parseInt(_0xb4efc4(420)) / 12);
      if (_0x214aa0 === _0x57ee56) break;
      else _0x1293dc["push"](_0x1293dc["shift"]());
    } catch (_0x4c6aa4) {
      _0x1293dc["push"](_0x1293dc["shift"]());
    }
  }
})(_0x441e, 316141);
class WebAppObserver {
  constructor(_0x2f618b) {
    var _a;
    const _0x37b039 = _0x4cf7, _0x32be59 = { "NFaWM": function(_0x4bc3e9, _0x310dbd) {
      return _0x4bc3e9 !== _0x310dbd;
    }, "zQXEo": _0x37b039(429) };
    this["webApp"] = _0x32be59[_0x37b039(430)](typeof window, _0x32be59[_0x37b039(422)]) && ((_a = window == null ? void 0 : window["Telegram"]) == null ? void 0 : _a["WebApp"]) ? window[_0x37b039(419)][_0x37b039(415)] : null, this[_0x37b039(418)] = _0x2f618b;
  }
  ["init"]() {
    var _a;
    const _0x576211 = _0x4cf7;
    if (this[_0x576211(414)] && ((_a = this[_0x576211(414)]) == null ? void 0 : _a[_0x576211(412)])) {
      const _0x49b1da = this[_0x576211(414)][_0x576211(412)];
      this[_0x576211(414)][_0x576211(412)] = (_0xa10756, _0x13e11c) => {
        const _0x2c0582 = _0x576211, _0x29c6b2 = _0xa10756[_0x2c0582(432)]("/")[_0x2c0582(428)]() || "";
        return this[_0x2c0582(418)][_0x2c0582(417)](Events[_0x2c0582(433)], { "slug": _0x29c6b2 }), _0x49b1da["call"](this[_0x2c0582(414)], _0xa10756, _0x13e11c);
      };
    }
  }
}
function _0x4cf7(_0x57b423, _0x31344a) {
  const _0x441e7d = _0x441e();
  return _0x4cf7 = function(_0x4cf7fa, _0x41a8c8) {
    _0x4cf7fa = _0x4cf7fa - 410;
    let _0x37bcab = _0x441e7d[_0x4cf7fa];
    return _0x37bcab;
  }, _0x4cf7(_0x57b423, _0x31344a);
}
function _0x441e() {
  const _0x2d7b2c = ["826VTnkPy", "30684IyHlbJ", "99748Vfsrgl", "88TNgMaI", "pop", "undefined", "NFaWM", "2791800odAvBm", "split", "PURCHASE_INIT", "583138WclNSB", "8bubBnZ", "openInvoice", "4566249LOuwqW", "webApp", "WebApp", "3yoFCLq", "collectEvent", "analyticsController", "Telegram", "1647312FvsvUz", "65IZVOEb", "zQXEo", "649206ZVSUAK"];
  _0x441e = function() {
    return _0x2d7b2c;
  };
  return _0x441e();
}
const _0x36d57c = _0x2f02;
(function(_0x2aa415, _0x2dbd8d) {
  const _0x45e9e4 = _0x2f02, _0x2bd577 = _0x2aa415();
  while (!![]) {
    try {
      const _0x51f63c = -parseInt(_0x45e9e4(367)) / 1 * (-parseInt(_0x45e9e4(371)) / 2) + parseInt(_0x45e9e4(342)) / 3 * (parseInt(_0x45e9e4(352)) / 4) + -parseInt(_0x45e9e4(346)) / 5 + parseInt(_0x45e9e4(377)) / 6 + parseInt(_0x45e9e4(368)) / 7 + -parseInt(_0x45e9e4(372)) / 8 * (-parseInt(_0x45e9e4(373)) / 9) + parseInt(_0x45e9e4(343)) / 10 * (-parseInt(_0x45e9e4(355)) / 11);
      if (_0x51f63c === _0x2dbd8d) break;
      else _0x2bd577["push"](_0x2bd577["shift"]());
    } catch (_0x16377e) {
      _0x2bd577["push"](_0x2bd577["shift"]());
    }
  }
})(_0x1b7b, 354775);
function _0x2f02(_0x5c4834, _0x20b33e) {
  const _0x1b7b50 = _0x1b7b();
  return _0x2f02 = function(_0x2f028c, _0x422591) {
    _0x2f028c = _0x2f028c - 341;
    let _0x5d8fb6 = _0x1b7b50[_0x2f028c];
    return _0x5d8fb6;
  }, _0x2f02(_0x5c4834, _0x20b33e);
}
function _0x1b7b() {
  const _0x21cb44 = ["214BFNJaL", "103336jlGsiN", "135kWtBnl", "addEventListener", "collectEvent", "Telegram", "4143480UEombp", "handleEvents", "WebView", "461307vixpdM", "10sZSdxS", "eventStatusMap", "zoeeC", "1429890MaJnmE", "receiveEvent", "status", "ZUFeY", "pLSPB", "TelegramGameProxy", "8uZpSKg", "CdXSh", "call", "12223167XwkuZZ", "analyticsController", "egZpG", "TelegramGameProxy_receiveEvent", "LXSJm", "PJAHH", "parse", "PURCHASE_FAILED", "PURCHASE_SUCCESS", "init", "hJYar", "PURCHASE_CANCELLED", "219DDeUzM", "3756508spoakt", "slug", "suwDq"];
  _0x1b7b = function() {
    return _0x21cb44;
  };
  return _0x1b7b();
}
class WebViewObserver {
  constructor(_0x5934fa) {
    const _0x50691a = _0x2f02;
    this[_0x50691a(344)] = { "paid": Events[_0x50691a(363)], "cancelled": Events[_0x50691a(366)], "failed": Events[_0x50691a(362)] }, this[_0x50691a(356)] = _0x5934fa;
  }
  [_0x36d57c(364)]() {
    const _0x194744 = _0x36d57c, _0x166b09 = { "ZUFeY": function(_0x4396d2, _0x315ba4) {
      return _0x4396d2 !== _0x315ba4;
    }, "pLSPB": _0x194744(345), "suwDq": _0x194744(360), "egZpG": "message" };
    window[_0x194744(374)](_0x166b09[_0x194744(357)], ({ data: _0x4fb7f0 }) => {
      const _0x35a967 = _0x194744;
      if (_0x166b09[_0x35a967(349)](_0x166b09[_0x35a967(350)], _0x166b09[_0x35a967(370)])) try {
        const { eventType: _0x457436, eventData: _0x4887c1 } = JSON[_0x35a967(361)](_0x4fb7f0);
        this[_0x35a967(378)](_0x457436, _0x4887c1);
      } catch (_0x5765ca) {
      }
      else _0x5c2933 = _0xe1c09f["receiveEvent"];
    }), this["handlePlatformListener"](window[_0x194744(351)]), this["handlePlatformListener"](window[_0x194744(376)][_0x194744(341)]), this["handlePlatformListener"](window[_0x194744(358)]);
  }
  ["handlePlatformListener"](_0x217311) {
    const _0x1fa2d0 = _0x36d57c, _0x8ed8b0 = { "CbXMc": function(_0xf4b2ad, _0x5c245b) {
      return _0xf4b2ad === _0x5c245b;
    }, "hJYar": _0x1fa2d0(359) };
    if (!_0x217311) return;
    let _0x3c4e0b;
    (_0x217311 == null ? void 0 : _0x217311[_0x1fa2d0(347)]) ? _0x3c4e0b = _0x217311[_0x1fa2d0(347)] : (_0x3c4e0b = _0x217311, _0x217311 = window);
    const _0x2ea59f = this;
    _0x217311[_0x1fa2d0(347)] = (_0x1cacfe, _0x338fd2) => {
      const _0x8b06c2 = _0x1fa2d0;
      return _0x8ed8b0["CbXMc"](_0x8ed8b0[_0x8b06c2(365)], _0x8b06c2(359)) ? (_0x2ea59f["handleEvents"](_0x1cacfe, _0x338fd2), _0x3c4e0b["call"](_0x217311, _0x1cacfe, _0x338fd2)) : (_0x42918f[_0x8b06c2(378)](_0x4fc2dd, _0x7c7cf1), _0x3fa954[_0x8b06c2(354)](_0x36a1eb, _0xd64442, _0x3d60da));
    };
  }
  [_0x36d57c(378)](_0x48220f, _0x5c67f8) {
    const _0x558bd2 = _0x36d57c, _0x188b55 = { "rKavg": function(_0x3f1510, _0x262e70) {
      return _0x3f1510 === _0x262e70;
    }, "CdXSh": "invoice_closed" };
    _0x188b55["rKavg"](_0x48220f, _0x188b55[_0x558bd2(353)]) && (this[_0x558bd2(344)][_0x5c67f8[_0x558bd2(348)]] && this[_0x558bd2(356)][_0x558bd2(375)](this[_0x558bd2(344)][_0x5c67f8[_0x558bd2(348)]], { "slug": _0x5c67f8[_0x558bd2(369)] }));
  }
}
function _0x3556() {
  var _0x1eb9ce = ["592458EsqlSM", "recordEvent", "init", "collectTappsEvent", "2|1|3|4|5|0", "441030HRABsy", "STG", "6uvXQKV", "VcmGJ", "sEsab", "36272wEkaFh", "tappsObserver", "mXTii", "split", "741msAyLH", "jbxTB", "lPcpu", "2557471WPebbs", "AOqDZ", "5711420lhpTZu", "UxraU", "catch", "collectEvent", "AJWmT", "914OUJaKS", "15VlcBxQ", "tonConnectObserver", "rYrIi", "error", "35044LMouqH", "json", "webViewObserver", "webAppObserver", "events/threshold", "documentObserver", "WaPum", "YxfJR", "eXztc", "eventsThreshold", "appModule", "369luKLJo"];
  _0x3556 = function() {
    return _0x1eb9ce;
  };
  return _0x3556();
}
var _0x3a56a4 = _0x1aaf;
(function(_0x22e216, _0x58bd11) {
  var _0x1ab9b5 = _0x1aaf, _0x3db4ca = _0x22e216();
  while (!![]) {
    try {
      var _0x32d192 = parseInt(_0x1ab9b5(304)) / 1 * (parseInt(_0x1ab9b5(314)) / 2) + -parseInt(_0x1ab9b5(331)) / 3 + -parseInt(_0x1ab9b5(319)) / 4 * (parseInt(_0x1ab9b5(315)) / 5) + -parseInt(_0x1ab9b5(297)) / 6 * (-parseInt(_0x1ab9b5(307)) / 7) + parseInt(_0x1ab9b5(300)) / 8 * (parseInt(_0x1ab9b5(330)) / 9) + parseInt(_0x1ab9b5(336)) / 10 + -parseInt(_0x1ab9b5(309)) / 11;
      if (_0x32d192 === _0x58bd11) break;
      else _0x3db4ca["push"](_0x3db4ca["shift"]());
    } catch (_0x111338) {
      _0x3db4ca["push"](_0x3db4ca["shift"]());
    }
  }
})(_0x3556, 190998);
function _0x1aaf(_0x4c36bb, _0x59088d) {
  var _0x35568a = _0x3556();
  return _0x1aaf = function(_0x1aaf00, _0x32c5d1) {
    _0x1aaf00 = _0x1aaf00 - 296;
    var _0x44f84f = _0x35568a[_0x1aaf00];
    return _0x44f84f;
  }, _0x1aaf(_0x4c36bb, _0x59088d);
}
class AnalyticsController {
  constructor(_0x542fff) {
    var _0x89a6f9 = _0x1aaf, _0xde5693 = { "AJWmT": _0x89a6f9(335) }, _0x8d52c3 = _0xde5693[_0x89a6f9(313)][_0x89a6f9(303)]("|"), _0x1b9064 = 0;
    while (!![]) {
      switch (_0x8d52c3[_0x1b9064++]) {
        case "0":
          this[_0x89a6f9(301)] = new TappsObserver(this);
          continue;
        case "1":
          this["documentObserver"] = new DocumentObserver(this);
          continue;
        case "2":
          this[_0x89a6f9(329)] = _0x542fff;
          continue;
        case "3":
          this[_0x89a6f9(316)] = new TonConnectObserver(this);
          continue;
        case "4":
          this["webAppObserver"] = new WebAppObserver(this);
          continue;
        case "5":
          this["webViewObserver"] = new WebViewObserver(this);
          continue;
      }
      break;
    }
  }
  async [_0x3a56a4(333)]() {
    var _0x3694d8 = _0x3a56a4, _0x4c551f = { "mXTii": function(_0x2e4ae9, _0x5839ce) {
      return _0x2e4ae9 + _0x5839ce;
    }, "sEsab": function(_0x6b725c, _0x46616b) {
      return _0x6b725c === _0x46616b;
    }, "pbkSH": _0x3694d8(296), "eXztc": _0x3694d8(323), "UxraU": function(_0x42ace8, _0x139e62) {
      return _0x42ace8 !== _0x139e62;
    }, "YkaaK": _0x3694d8(306), "jbxTB": _0x3694d8(325) };
    this["documentObserver"][_0x3694d8(333)](), this[_0x3694d8(316)]["init"](), this[_0x3694d8(322)][_0x3694d8(333)](), this[_0x3694d8(321)][_0x3694d8(333)](), this["tappsObserver"][_0x3694d8(333)]();
    try {
      this[_0x3694d8(328)] = await (await fetch(_0x4c551f[_0x3694d8(302)](_0x4c551f[_0x3694d8(299)](this[_0x3694d8(329)]["env"], _0x4c551f["pbkSH"]) ? STAGING_BACKEND_URL : BACKEND_URL, _0x4c551f[_0x3694d8(327)]), { "signal": AbortSignal["timeout"](2e3) }))[_0x3694d8(320)]();
    } catch (_0x135837) {
      _0x4c551f[_0x3694d8(310)](_0x4c551f["YkaaK"], _0x4c551f[_0x3694d8(305)]) ? this["eventsThreshold"] = { "app-hide": 3 } : this[_0x3694d8(328)] = { "app-hide": 3 };
    }
  }
  [_0x3a56a4(332)](_0x43dafe, _0x36ae26) {
    var _0x17ffb4 = _0x3a56a4;
    this[_0x17ffb4(329)]["recordEvent"](_0x43dafe, _0x36ae26)[_0x17ffb4(311)]((_0x521cf7) => console[_0x17ffb4(318)](_0x521cf7));
  }
  [_0x3a56a4(312)](_0x450083, _0x56c02b) {
    var _0x5d1be8 = _0x3a56a4, _0x23bb86 = { "YxfJR": function(_0xa937d4, _0x17dc9f) {
      return _0xa937d4 === _0x17dc9f;
    }, "PrnGy": function(_0x37d2a7, _0x2de10e) {
      return _0x37d2a7 !== _0x2de10e;
    }, "rYrIi": _0x5d1be8(308), "VcmGJ": "Acrft" };
    if (_0x23bb86[_0x5d1be8(326)](this[_0x5d1be8(328)][_0x450083], 0)) {
      if (_0x23bb86["PrnGy"](_0x23bb86[_0x5d1be8(317)], _0x23bb86[_0x5d1be8(298)])) return;
      else this["appModule"] = _0x6a35d1, this[_0x5d1be8(324)] = new _0x17bbda(this), this[_0x5d1be8(316)] = new _0x5a203a(this), this[_0x5d1be8(322)] = new _0x3a55fa(this), this[_0x5d1be8(321)] = new _0x2728df(this), this["tappsObserver"] = new _0x14412e(this);
    }
    this["appModule"][_0x5d1be8(312)](_0x450083, _0x56c02b), this[_0x5d1be8(328)][_0x450083] && this[_0x5d1be8(328)][_0x450083]--;
  }
  [_0x3a56a4(334)](_0x436b84, _0x47c29e) {
    var _0x3c7cff = _0x3a56a4;
    this["appModule"][_0x3c7cff(334)](_0x436b84, _0x47c29e);
  }
}
(function(_0xc86118, _0x453a10) {
  const _0x3534c0 = _0x1ee2, _0x5d635a = _0xc86118();
  while (!![]) {
    try {
      const _0x1e7f73 = -parseInt(_0x3534c0(236)) / 1 * (parseInt(_0x3534c0(241)) / 2) + parseInt(_0x3534c0(240)) / 3 * (-parseInt(_0x3534c0(234)) / 4) + -parseInt(_0x3534c0(243)) / 5 + parseInt(_0x3534c0(244)) / 6 * (-parseInt(_0x3534c0(228)) / 7) + parseInt(_0x3534c0(233)) / 8 * (-parseInt(_0x3534c0(227)) / 9) + -parseInt(_0x3534c0(239)) / 10 * (-parseInt(_0x3534c0(232)) / 11) + -parseInt(_0x3534c0(230)) / 12 * (-parseInt(_0x3534c0(237)) / 13);
      if (_0x1e7f73 === _0x453a10) break;
      else _0x5d635a["push"](_0x5d635a["shift"]());
    } catch (_0x1d9456) {
      _0x5d635a["push"](_0x5d635a["shift"]());
    }
  }
})(_0x23ba, 272967);
const throwError = (_0x14a757) => {
  throw new Error(_0x14a757);
};
function _0x1ee2(_0x4dae5d, _0x3a7417) {
  const _0x23ba1b = _0x23ba();
  return _0x1ee2 = function(_0x1ee230, _0x2cbe5e) {
    _0x1ee230 = _0x1ee230 - 226;
    let _0x315a78 = _0x23ba1b[_0x1ee230];
    return _0x315a78;
  }, _0x1ee2(_0x4dae5d, _0x3a7417);
}
var Errors = ((_0x1ad7c6) => {
  const _0xc1d4e2 = _0x1ee2, _0xcb9404 = { "sDpEy": _0xc1d4e2(229), "kLFBN": "USER_DATA_IS_NOT_PROVIDED", "AzymU": _0xc1d4e2(231) };
  return _0x1ad7c6[_0xc1d4e2(235)] = _0xcb9404[_0xc1d4e2(226)], _0x1ad7c6[_0xcb9404[_0xc1d4e2(238)]] = _0xcb9404[_0xc1d4e2(242)], _0x1ad7c6;
})(Errors || {});
function _0x23ba() {
  const _0x5bb6b6 = ["Token is not provided.", "276ygMvDW", "Telegram User data is not provided.", "44ccETHj", "1484008BEaHOd", "4XksVcx", "TOKEN_IS_NOT_PROVIDED", "1ibfcqD", "852553YIOppn", "kLFBN", "473470GFXMkB", "1446969OKZmkF", "113658VInNwL", "AzymU", "1071915ZqFqcq", "1801482AGHgZp", "sDpEy", "18uFFYkd", "7EATSmh"];
  _0x23ba = function() {
    return _0x5bb6b6;
  };
  return _0x23ba();
}
(function(_0x1f0d62, _0x2161be) {
  const _0x15e2fc = _0x4b6e, _0x5da060 = _0x1f0d62();
  while (!![]) {
    try {
      const _0x1dd352 = -parseInt(_0x15e2fc(147)) / 1 * (parseInt(_0x15e2fc(139)) / 2) + -parseInt(_0x15e2fc(134)) / 3 + -parseInt(_0x15e2fc(136)) / 4 * (parseInt(_0x15e2fc(143)) / 5) + -parseInt(_0x15e2fc(150)) / 6 + parseInt(_0x15e2fc(151)) / 7 * (-parseInt(_0x15e2fc(135)) / 8) + -parseInt(_0x15e2fc(137)) / 9 * (parseInt(_0x15e2fc(148)) / 10) + -parseInt(_0x15e2fc(141)) / 11 * (-parseInt(_0x15e2fc(149)) / 12);
      if (_0x1dd352 === _0x2161be) break;
      else _0x5da060["push"](_0x5da060["shift"]());
    } catch (_0x7aa0fb) {
      _0x5da060["push"](_0x5da060["shift"]());
    }
  }
})(_0x33a8, 716286);
function _0x4b6e(_0x50c058, _0x43bb94) {
  const _0x33a8d4 = _0x33a8();
  return _0x4b6e = function(_0x4b6e05, _0x24576f) {
    _0x4b6e05 = _0x4b6e05 - 134;
    let _0x194ada = _0x33a8d4[_0x4b6e05];
    return _0x194ada;
  }, _0x4b6e(_0x50c058, _0x43bb94);
}
function _0x33a8() {
  const _0x88e53e = ["8545884ZnOfnp", "2975KprMjp", "723639hKxszD", "8864DikTKc", "71620HtJeKn", "9rVplNm", "stringify", "806RbSvKU", "stream", "59644761TpELrG", "gzip", "290seSKiw", "blob", "YJKRb", "pipeThrough", "2843DxRMSG", "3853190oJSBRB", "12YaWBlc"];
  _0x33a8 = function() {
    return _0x88e53e;
  };
  return _0x33a8();
}
async function compressData(_0x1534d9) {
  const _0x3e3125 = _0x4b6e, _0x5312bd = { "YJKRb": _0x3e3125(142) }, _0x448792 = new Blob([JSON[_0x3e3125(138)](_0x1534d9)])[_0x3e3125(140)](), _0xf7e223 = _0x448792[_0x3e3125(146)](new CompressionStream(_0x5312bd[_0x3e3125(145)]));
  return await new Response(_0xf7e223)[_0x3e3125(144)]();
}
const _0x27177c = _0x41ec;
(function(_0x3584d2, _0x25c336) {
  const _0x4a380c = _0x41ec, _0x1e0586 = _0x3584d2();
  while (!![]) {
    try {
      const _0x3b96bb = -parseInt(_0x4a380c(454)) / 1 * (parseInt(_0x4a380c(471)) / 2) + -parseInt(_0x4a380c(472)) / 3 * (parseInt(_0x4a380c(491)) / 4) + parseInt(_0x4a380c(456)) / 5 + -parseInt(_0x4a380c(501)) / 6 * (-parseInt(_0x4a380c(487)) / 7) + -parseInt(_0x4a380c(475)) / 8 + -parseInt(_0x4a380c(462)) / 9 + parseInt(_0x4a380c(473)) / 10;
      if (_0x3b96bb === _0x25c336) break;
      else _0x1e0586["push"](_0x1e0586["shift"]());
    } catch (_0x3b19b9) {
      _0x1e0586["push"](_0x1e0586["shift"]());
    }
  }
})(_0x1611, 326980);
function _0x41ec(_0x295554, _0x4d43d0) {
  const _0x1611b7 = _0x1611();
  return _0x41ec = function(_0x41ec2c, _0xe7146d) {
    _0x41ec2c = _0x41ec2c - 453;
    let _0x29807d = _0x1611b7[_0x41ec2c];
    return _0x29807d;
  }, _0x41ec(_0x295554, _0x4d43d0);
}
function _0x1611() {
  const _0x17da2e = ["bsqKk", "xuXbx", "mwVgX", "Content", "appModule", "TOKEN_IS_NOT_PROVIDED", "2GqpVIw", "6BjetFL", "7832800nIVtnU", "ODkls", "2557192xqXdMc", "WRXVM", "nInDN", "UxsHf", "DTxCX", "events", "recordEvent", "GGinQ", "stringify", "env", "init", "hIXyV", "939862DYBOpM", "TQncR", "assign", "BACKEND_URL", "450392pCdDTN", "Ninop", "Content-Encoding", "fmUOP", "vKrdc", "gzip", "WSAaa", "wqomQ", "taskSolution", "clone", "18RwJJUS", "hSaHU", "GVRdy", "generateHeaders", "STG", "status", "POST", "rAMfA", "OImMX", "pIsMX", "qjqGo", "540330HtIUOg", "responseToParams", "2373975ctefUa", "custom_data", "application/json", "sPztM", "setNewArgs", "getApiToken", "2238462eAQyHO", "vxBMA", "json"];
  _0x1611 = function() {
    return _0x17da2e;
  };
  return _0x1611();
}
class NetworkController {
  constructor(_0x327c11) {
    const _0x3f055e = _0x41ec, _0x5f4e6b = { "nInDN": _0x3f055e(478), "KDwGT": function(_0x20a06a, _0x5a4263) {
      return _0x20a06a(_0x5a4263);
    }, "rAMfA": function(_0x1bedd4, _0x40ae56) {
      return _0x1bedd4 === _0x40ae56;
    }, "vxBMA": function(_0x409172, _0x34b6f9) {
      return _0x409172 === _0x34b6f9;
    }, "TQncR": "tHmqB", "WSAaa": _0x3f055e(468), "mwVgX": _0x3f055e(493), "pIsMX": "gzip", "qjqGo": _0x3f055e(458), "KbkLn": function(_0x16a86a, _0x59385b) {
      return _0x16a86a !== _0x59385b;
    }, "DTxCX": "OImMX", "Ninop": function(_0xa259fd, _0x3259a3) {
      return _0xa259fd === _0x3259a3;
    }, "vKrdc": function(_0x109d1f, _0x5ceace) {
      return _0x109d1f === _0x5ceace;
    }, "hSaHU": _0x3f055e(505) };
    this["BACKEND_URL"] = BACKEND_URL, this[_0x3f055e(455)] = async (_0x351ae6) => {
      const _0x10e5b9 = _0x3f055e, _0x1c2635 = { "RxZdG": function(_0x49eb21, _0x49354c) {
        return _0x49eb21(_0x49354c);
      } };
      if (_0x5f4e6b[_0x10e5b9(477)] === _0x5f4e6b["nInDN"]) {
        const _0x4b502c = _0x351ae6[_0x10e5b9(500)]();
        if (_0x5f4e6b["KDwGT"](String, _0x4b502c[_0x10e5b9(506)])[0] === "2" || _0x5f4e6b[_0x10e5b9(508)](_0x4b502c[_0x10e5b9(506)], 429)) {
          if (_0x5f4e6b[_0x10e5b9(463)](_0x5f4e6b[_0x10e5b9(488)], _0x5f4e6b[_0x10e5b9(488)])) {
            const _0x515a75 = await _0x4b502c[_0x10e5b9(464)]();
            this[_0x10e5b9(469)][_0x10e5b9(460)](_0x515a75[_0x5f4e6b[_0x10e5b9(497)]]);
          } else _0x1f0534[_0x10e5b9(493)] = _0x10e5b9(496);
        }
        return _0x351ae6;
      } else _0x1c2635["RxZdG"](_0x19c66b, _0xc53eeb["TOKEN_IS_NOT_PROVIDED"]);
    }, this[_0x3f055e(504)] = (_0x474498) => {
      const _0x1f4e94 = _0x3f055e;
      this[_0x1f4e94(469)]["solveTask"]();
      const _0x2b0167 = {};
      this["appModule"]["taskSolution"] && (_0x5f4e6b["KbkLn"](_0x5f4e6b[_0x1f4e94(479)], _0x1f4e94(509)) ? _0x194547 = _0x496e48[_0x1f4e94(489)](_0x11f38d[_0x1f4e94(457)], _0x53e0d7) : _0x2b0167[_0x1f4e94(468)] = this[_0x1f4e94(469)]["taskSolution"]);
      if (_0x474498) {
        if (_0x5f4e6b[_0x1f4e94(492)](_0x1f4e94(459), _0x1f4e94(494))) {
          this[_0x1f4e94(469)]["solveTask"]();
          const _0x73eabc = {};
          return this[_0x1f4e94(469)][_0x1f4e94(499)] && (_0x73eabc[_0x5f4e6b[_0x1f4e94(497)]] = this[_0x1f4e94(469)][_0x1f4e94(499)]), _0x29614b && (_0x73eabc[_0x5f4e6b[_0x1f4e94(467)]] = _0x5f4e6b[_0x1f4e94(510)]), { "TGA-Auth-Token": this["appModule"][_0x1f4e94(461)](), "Content-Type": _0x5f4e6b[_0x1f4e94(453)], ..._0x73eabc };
        } else _0x2b0167[_0x5f4e6b[_0x1f4e94(467)]] = _0x5f4e6b["pIsMX"];
      }
      return { "TGA-Auth-Token": this["appModule"][_0x1f4e94(461)](), "Content-Type": _0x5f4e6b[_0x1f4e94(453)], ..._0x2b0167 };
    }, this["appModule"] = _0x327c11, _0x5f4e6b[_0x3f055e(495)](this[_0x3f055e(469)][_0x3f055e(484)], _0x5f4e6b[_0x3f055e(502)]) && (this[_0x3f055e(490)] = STAGING_BACKEND_URL), !this[_0x3f055e(469)][_0x3f055e(461)]() && throwError(Errors[_0x3f055e(470)]);
  }
  [_0x27177c(485)]() {
  }
  async ["recordEvents"](_0x3c0a94, _0x33e69d = !![]) {
    const _0x2a1ced = _0x27177c, _0x1d66a7 = { "bsqKk": function(_0x37f082, _0x329257, _0xd45e00) {
      return _0x37f082(_0x329257, _0xd45e00);
    }, "WRXVM": function(_0x473a46, _0x30de4d) {
      return _0x473a46 + _0x30de4d;
    }, "GVRdy": _0x2a1ced(480), "EKtEl": _0x2a1ced(507) };
    return await _0x1d66a7[_0x2a1ced(465)](fetch, _0x1d66a7[_0x2a1ced(476)](this[_0x2a1ced(490)], _0x1d66a7[_0x2a1ced(503)]), { "method": _0x1d66a7["EKtEl"], "headers": this["generateHeaders"](_0x33e69d), "body": _0x33e69d ? await compressData(_0x3c0a94) : JSON[_0x2a1ced(483)](_0x3c0a94) })["then"](this[_0x2a1ced(455)], this[_0x2a1ced(455)]);
  }
  async [_0x27177c(481)](_0x555eae, _0x5d63f0, _0x31e4e0, _0x5c3c2d = !![]) {
    const _0x25933e = _0x27177c, _0x699f3f = { "ODkls": function(_0x554f03, _0x2a4051) {
      return _0x554f03 !== _0x2a4051;
    }, "hIXyV": _0x25933e(498), "GGinQ": function(_0x5242f1, _0x338011) {
      return _0x5242f1 + _0x338011;
    }, "MbxNk": _0x25933e(507), "xuXbx": function(_0x1947e9, _0x20e829) {
      return _0x1947e9(_0x20e829);
    } };
    (_0x5d63f0 == null ? void 0 : _0x5d63f0[_0x25933e(457)]) && (!_0x31e4e0 ? _0x699f3f[_0x25933e(474)](_0x699f3f[_0x25933e(486)], _0x699f3f[_0x25933e(486)]) ? !_0x5a80ab ? _0x25d59f = _0x48795b["custom_data"] : _0x4aa7bd = _0x13b633[_0x25933e(489)](_0x4c2b46["custom_data"], _0x274031) : _0x31e4e0 = _0x5d63f0[_0x25933e(457)] : _0x31e4e0 = Object[_0x25933e(489)](_0x5d63f0[_0x25933e(457)], _0x31e4e0));
    const _0x592b76 = { ..._0x5d63f0, "event_name": _0x555eae, "custom_data": _0x31e4e0, ...this[_0x25933e(469)]["assembleEventSession"]() };
    await fetch(_0x699f3f[_0x25933e(482)](this[_0x25933e(490)], "events"), { "method": _0x699f3f["MbxNk"], "headers": this[_0x25933e(504)](!![]), "body": _0x5c3c2d ? await _0x699f3f[_0x25933e(466)](compressData, _0x592b76) : JSON[_0x25933e(483)](_0x592b76) })["then"](this[_0x25933e(455)], this[_0x25933e(455)]);
  }
}
class V extends Error {
  constructor(e, t, n) {
    super(t, { cause: n }), this.type = e, Object.setPrototypeOf(this, V.prototype);
  }
}
function f(s, e, t) {
  return new V(s, e, t);
}
const Ye = "ERR_UNEXPECTED_TYPE", ce = "ERR_PARSE";
function E() {
  return f(Ye, "Value has unexpected type");
}
class D {
  constructor(e, t, n) {
    this.parser = e, this.isOptional = t, this.type = n;
  }
  /**
   * Attempts to parse passed value
   * @param value - value to parse.
   * @throws {SDKError} ERR_PARSE
   * @see ERR_PARSE
   */
  parse(e) {
    if (!(this.isOptional && e === void 0))
      try {
        return this.parser(e);
      } catch (t) {
        throw f(
          ce,
          `Unable to parse value${this.type ? ` as ${this.type}` : ""}`,
          t
        );
      }
  }
  optional() {
    return this.isOptional = true, this;
  }
}
function S(s, e) {
  return () => new D(s, false, e);
}
const b = S((s) => {
  if (typeof s == "boolean")
    return s;
  const e = String(s);
  if (e === "1" || e === "true")
    return true;
  if (e === "0" || e === "false")
    return false;
  throw E();
}, "boolean");
function pe(s, e) {
  const t = {};
  for (const n in s) {
    const r = s[n];
    if (!r)
      continue;
    let i, o;
    if (typeof r == "function" || "parse" in r)
      i = n, o = typeof r == "function" ? r : r.parse.bind(r);
    else {
      const { type: a } = r;
      i = r.from || n, o = typeof a == "function" ? a : a.parse.bind(a);
    }
    try {
      const a = o(e(i));
      a !== void 0 && (t[n] = a);
    } catch (a) {
      throw f(ce, `Unable to parse field "${n}"`, a);
    }
  }
  return t;
}
function he(s) {
  let e = s;
  if (typeof e == "string" && (e = JSON.parse(e)), typeof e != "object" || e === null || Array.isArray(e))
    throw E();
  return e;
}
function g(s, e) {
  return new D((t) => {
    const n = he(t);
    return pe(s, (r) => n[r]);
  }, false, e);
}
const y = S((s) => {
  if (typeof s == "number")
    return s;
  if (typeof s == "string") {
    const e = Number(s);
    if (!Number.isNaN(e))
      return e;
  }
  throw E();
}, "number"), h = S((s) => {
  if (typeof s == "string" || typeof s == "number")
    return s.toString();
  throw E();
}, "string");
({
  custom_method_invoked: g({
    req_id: h(),
    result: (s) => s,
    error: h().optional()
  })
});
function Q(s) {
  return /^#[\da-f]{6}$/i.test(s);
}
function ht(s) {
  return /^#[\da-f]{3}$/i.test(s);
}
function me(s) {
  const e = s.replace(/\s/g, "").toLowerCase();
  if (Q(e))
    return e;
  if (ht(e)) {
    let n = "#";
    for (let r = 0; r < 3; r += 1)
      n += e[1 + r].repeat(2);
    return n;
  }
  const t = e.match(/^rgb\((\d{1,3}),(\d{1,3}),(\d{1,3})\)$/) || e.match(/^rgba\((\d{1,3}),(\d{1,3}),(\d{1,3}),\d{1,3}\)$/);
  if (!t)
    throw new Error(`Value "${s}" does not satisfy any of known RGB formats.`);
  return t.slice(1).reduce((n, r) => {
    const i = parseInt(r, 10).toString(16);
    return n + (i.length === 1 ? "0" : "") + i;
  }, "#");
}
const Ee = S((s) => s instanceof Date ? s : new Date(y().parse(s) * 1e3), "Date");
function K(s, e) {
  return new D((t) => {
    if (typeof t != "string" && !(t instanceof URLSearchParams))
      throw E();
    const n = typeof t == "string" ? new URLSearchParams(t) : t;
    return pe(s, (r) => {
      const i = n.get(r);
      return i === null ? void 0 : i;
    });
  }, false, e);
}
const dt = g({
  id: y(),
  type: h(),
  title: h(),
  photoUrl: {
    type: h().optional(),
    from: "photo_url"
  },
  username: h().optional()
}, "Chat").optional(), ne = g({
  addedToAttachmentMenu: {
    type: b().optional(),
    from: "added_to_attachment_menu"
  },
  allowsWriteToPm: {
    type: b().optional(),
    from: "allows_write_to_pm"
  },
  firstName: {
    type: h(),
    from: "first_name"
  },
  id: y(),
  isBot: {
    type: b().optional(),
    from: "is_bot"
  },
  isPremium: {
    type: b().optional(),
    from: "is_premium"
  },
  languageCode: {
    type: h().optional(),
    from: "language_code"
  },
  lastName: {
    type: h().optional(),
    from: "last_name"
  },
  photoUrl: {
    type: h().optional(),
    from: "photo_url"
  },
  username: h().optional()
}, "User").optional();
function Se() {
  return K({
    authDate: {
      type: Ee(),
      from: "auth_date"
    },
    canSendAfter: {
      type: y().optional(),
      from: "can_send_after"
    },
    chat: dt,
    chatInstance: {
      type: h().optional(),
      from: "chat_instance"
    },
    chatType: {
      type: h().optional(),
      from: "chat_type"
    },
    hash: h(),
    queryId: {
      type: h().optional(),
      from: "query_id"
    },
    receiver: ne,
    startParam: {
      type: h().optional(),
      from: "start_param"
    },
    user: ne
  }, "InitData");
}
const _t = S((s) => me(h().parse(s)), "rgb");
function ft(s) {
  return s.replace(/_[a-z]/g, (e) => e[1].toUpperCase());
}
function gt(s) {
  return s.replace(/[A-Z]/g, (e) => `_${e.toLowerCase()}`);
}
const Pe = S(
  (s) => {
    const e = _t().optional();
    return Object.entries(he(s)).reduce((t, [n, r]) => (t[ft(n)] = e.parse(r), t), {});
  },
  "ThemeParams"
);
function X(s) {
  return K({
    botInline: {
      type: b().optional(),
      from: "tgWebAppBotInline"
    },
    initData: {
      type: Se().optional(),
      from: "tgWebAppData"
    },
    initDataRaw: {
      type: h().optional(),
      from: "tgWebAppData"
    },
    platform: {
      type: h(),
      from: "tgWebAppPlatform"
    },
    showSettings: {
      type: b().optional(),
      from: "tgWebAppShowSettings"
    },
    startParam: {
      type: h().optional(),
      from: "tgWebAppStartParam"
    },
    themeParams: {
      type: Pe(),
      from: "tgWebAppThemeParams"
    },
    version: {
      type: h(),
      from: "tgWebAppVersion"
    }
  }).parse(s);
}
function xe(s) {
  return X(
    s.replace(/^[^?#]*[?#]/, "").replace(/[?#]/g, "&")
  );
}
function wt() {
  return xe(window.location.href);
}
function Ce() {
  return performance.getEntriesByType("navigation")[0];
}
function bt() {
  const s = Ce();
  if (!s)
    throw new Error("Unable to get first navigation entry.");
  return xe(s.name);
}
function Te(s) {
  return `telegram-apps/${s.replace(/[A-Z]/g, (e) => `-${e.toLowerCase()}`)}`;
}
function Re(s, e) {
  sessionStorage.setItem(Te(s), JSON.stringify(e));
}
function Ae(s) {
  const e = sessionStorage.getItem(Te(s));
  try {
    return e ? JSON.parse(e) : void 0;
  } catch {
  }
}
function mt() {
  return X(Ae("launchParams") || "");
}
function Ie(s) {
  return JSON.stringify(
    Object.fromEntries(
      Object.entries(s).map(([e, t]) => [gt(e), t])
    )
  );
}
function yt(s) {
  const {
    initDataRaw: e,
    themeParams: t,
    platform: n,
    version: r,
    showSettings: i,
    startParam: o,
    botInline: a
  } = s, p = new URLSearchParams();
  return p.set("tgWebAppPlatform", n), p.set("tgWebAppThemeParams", Ie(t)), p.set("tgWebAppVersion", r), e && p.set("tgWebAppData", e), o && p.set("tgWebAppStartParam", o), typeof i == "boolean" && p.set("tgWebAppShowSettings", i ? "1" : "0"), typeof a == "boolean" && p.set("tgWebAppBotInline", a ? "1" : "0"), p.toString();
}
function qe(s) {
  Re("launchParams", yt(s));
}
function vt() {
  const s = [];
  for (const e of [
    // Try to retrieve launch parameters from the current location. This method can return
    // nothing in case, location was changed, and then the page was reloaded.
    wt,
    // Then, try using the lower level API - window.performance.
    bt,
    // Finally, try to extract launch parameters from the session storage.
    mt
  ])
    try {
      const t = e();
      return qe(t), t;
    } catch (t) {
      s.push(t instanceof Error ? t.message : JSON.stringify(t));
    }
  throw new Error([
    `Unable to retrieve launch parameters from any known source. Perhaps, you have opened your app outside Telegram?
`,
    "📖 Refer to docs for more information:",
    `https://docs.telegram-mini-apps.com/packages/telegram-apps-sdk/environment
`,
    "Collected errors:",
    s.map((e) => `— ${e}`)
  ].join(`
`));
}
(function(_0xdf8f18, _0x3f075c) {
  const _0x1509f4 = _0x28ee, _0x3259a0 = _0xdf8f18();
  while (!![]) {
    try {
      const _0x7ca676 = parseInt(_0x1509f4(195)) / 1 * (parseInt(_0x1509f4(205)) / 2) + parseInt(_0x1509f4(206)) / 3 * (-parseInt(_0x1509f4(199)) / 4) + -parseInt(_0x1509f4(216)) / 5 + parseInt(_0x1509f4(202)) / 6 * (parseInt(_0x1509f4(214)) / 7) + -parseInt(_0x1509f4(198)) / 8 * (parseInt(_0x1509f4(207)) / 9) + -parseInt(_0x1509f4(212)) / 10 + parseInt(_0x1509f4(196)) / 11;
      if (_0x7ca676 === _0x3f075c) break;
      else _0x3259a0["push"](_0x3259a0["shift"]());
    } catch (_0x4ad9a9) {
      _0x3259a0["push"](_0x3259a0["shift"]());
    }
  }
})(_0x4143, 990185);
function _0x4143() {
  const _0x486241 = ["tCgnp", "LpFeR", "958QjJgGo", "297264YngjXT", "1055556HVmMPM", "getTime", "mIswe", "ouqVW", "xJVIc", "15404430ECoFCV", "tFCVY", "2415NjPrkW", "floor", "6137680TKEEuG", "toString", "length", "409bufKQu", "62590660GfTYnj", "dCvGe", "80RwgLKK", "76oowHOy", "KXTbu", "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx", "16134orLVsm"];
  _0x4143 = function() {
    return _0x486241;
  };
  return _0x4143();
}
function _0x28ee(_0xbcd351, _0x2cfd99) {
  const _0x4143a2 = _0x4143();
  return _0x28ee = function(_0x28ee39, _0x39d157) {
    _0x28ee39 = _0x28ee39 - 193;
    let _0x28b493 = _0x4143a2[_0x28ee39];
    return _0x28b493;
  }, _0x28ee(_0xbcd351, _0x2cfd99);
}
function generateUUID(_0x4b43de) {
  const _0x2b82f1 = _0x28ee, _0x575599 = { "dCvGe": function(_0x205da1, _0x4bbe00) {
    return _0x205da1 < _0x4bbe00;
  }, "mIswe": function(_0x1f9e3a, _0x122775) {
    return _0x1f9e3a + _0x122775;
  }, "tFCVY": function(_0x45ba5a, _0x1320b6) {
    return _0x45ba5a - _0x1320b6;
  }, "xJVIc": function(_0x266ce1, _0x349d79) {
    return _0x266ce1 << _0x349d79;
  }, "ouqVW": function(_0x53829f, _0xa552d8) {
    return _0x53829f | _0xa552d8;
  }, "LpFeR": function(_0x65d343, _0x535e50, _0x20aec9) {
    return _0x65d343(_0x535e50, _0x20aec9);
  }, "tCgnp": function(_0x17e884, _0x314e56) {
    return _0x17e884 / _0x314e56;
  }, "MpWpO": function(_0x1d4041, _0x45f8fc) {
    return _0x1d4041 & _0x45f8fc;
  }, "KXTbu": function(_0x3e5191, _0x22d1de) {
    return _0x3e5191 < _0x22d1de;
  } };
  let _0x1b6ebc = _0x2b82f1(201), _0x16b66d = (/* @__PURE__ */ new Date())[_0x2b82f1(208)]();
  const _0x10cee8 = _0x4b43de + _0x16b66d;
  function _0x2555c5(_0x447cbb) {
    const _0x4ddf19 = _0x2b82f1;
    let _0x41f51f = 0;
    for (let _0x419e63 = 0; _0x575599[_0x4ddf19(197)](_0x419e63, _0x447cbb[_0x4ddf19(194)]); _0x419e63++) {
      const _0x2c6ea = _0x447cbb["charCodeAt"](_0x419e63);
      _0x41f51f = _0x575599[_0x4ddf19(209)](_0x575599[_0x4ddf19(213)](_0x575599[_0x4ddf19(211)](_0x41f51f, 5), _0x41f51f), _0x2c6ea), _0x41f51f |= 0;
    }
    return _0x41f51f;
  }
  let _0x5ba081 = _0x2555c5(_0x10cee8)[_0x2b82f1(193)](16);
  while (_0x575599[_0x2b82f1(200)](_0x5ba081[_0x2b82f1(194)], 32)) {
    _0x5ba081 += _0x5ba081;
  }
  _0x5ba081 = _0x5ba081["slice"](0, 32);
  let _0x389cda = 0;
  return _0x1b6ebc["replace"](/[xy]/g, function(_0x1d73a9) {
    const _0x5c306a = _0x2b82f1, _0x5927c6 = _0x575599[_0x5c306a(210)](_0x575599[_0x5c306a(209)](_0x16b66d, _0x575599[_0x5c306a(204)](parseInt, _0x5ba081[_0x389cda], 16)) % 16, 0);
    return _0x16b66d = Math[_0x5c306a(215)](_0x575599[_0x5c306a(203)](_0x16b66d, 16)), _0x389cda++, (_0x1d73a9 == "x" ? _0x5927c6 : _0x575599[_0x5c306a(210)](_0x575599["MpWpO"](_0x5927c6, 3), 8))[_0x5c306a(193)](16);
  });
}
const _0x34f128 = _0x47eb;
(function(_0x2736c5, _0x3b5b49) {
  const _0x3bc8b1 = _0x47eb, _0x221bf4 = _0x2736c5();
  while (!![]) {
    try {
      const _0x4ecffe = parseInt(_0x3bc8b1(258)) / 1 + parseInt(_0x3bc8b1(233)) / 2 * (parseInt(_0x3bc8b1(246)) / 3) + parseInt(_0x3bc8b1(237)) / 4 + -parseInt(_0x3bc8b1(257)) / 5 * (parseInt(_0x3bc8b1(241)) / 6) + parseInt(_0x3bc8b1(232)) / 7 * (-parseInt(_0x3bc8b1(252)) / 8) + -parseInt(_0x3bc8b1(236)) / 9 + parseInt(_0x3bc8b1(231)) / 10;
      if (_0x4ecffe === _0x3b5b49) break;
      else _0x221bf4["push"](_0x221bf4["shift"]());
    } catch (_0x145464) {
      _0x221bf4["push"](_0x221bf4["shift"]());
    }
  }
})(_0x1cca, 175610);
function _0x1cca() {
  const _0x1aff4c = ["3804960QzDonk", "1365tySAfN", "34018FIhJqH", "userId", "QrozI", "2909232SZinlY", "152864lXpSkZ", "getUserIsPremium", "getPlatform", "username", "81570jJmmIh", "initData", "getWebAppStartParam", "sessionId", "userLocale", "57ThCYEQ", "startParam", "languageCode", "isPremium", "lastName", "userData", "13448AIkidm", "appModule", "is_premium", "getUserData", "getAppName", "85tJzlWi", "315885oQBUlK", "WJizR", "fiFAH", "USER_DATA_IS_NOT_PROVIDED", "firstName", "tNuCP", "gFBNL", "vpqwL", "platform", "assembleEventSession", "getUserLocale", "getUserId", "init", "webAppStartParam", "now", "bHqIX", "isBot"];
  _0x1cca = function() {
    return _0x1aff4c;
  };
  return _0x1cca();
}
function _0x47eb(_0x1bb0e6, _0x26a7d5) {
  const _0x1cca1c = _0x1cca();
  return _0x47eb = function(_0x47eba5, _0x26527f) {
    _0x47eba5 = _0x47eba5 - 226;
    let _0x50fb99 = _0x1cca1c[_0x47eba5];
    return _0x50fb99;
  }, _0x47eb(_0x1bb0e6, _0x26a7d5);
}
class SessionController {
  constructor(_0x50d3f4) {
    const _0x30a3f9 = _0x47eb;
    this[_0x30a3f9(253)] = _0x50d3f4;
  }
  [_0x34f128(226)]() {
    var _a;
    const _0x23eb2c = _0x34f128, _0x408df5 = { "WJizR": function(_0x5cb659) {
      return _0x5cb659();
    }, "bHqIX": function(_0x3e0e3d, _0x2130cf) {
      return _0x3e0e3d !== _0x2130cf;
    }, "tNuCP": "gFBNL", "QrozI": function(_0xe573e5, _0x1c8cc8) {
      return _0xe573e5(_0x1c8cc8);
    } }, _0x258462 = _0x408df5[_0x23eb2c(259)](vt), _0x24cdcd = _0x258462[_0x23eb2c(242)], _0x3910ac = (_a = _0x258462[_0x23eb2c(242)]) == null ? void 0 : _a["user"];
    if (!_0x3910ac) {
      if (_0x408df5[_0x23eb2c(229)](_0x408df5[_0x23eb2c(263)], _0x23eb2c(264))) return this[_0x23eb2c(244)];
      else _0x408df5[_0x23eb2c(235)](throwError, Errors[_0x23eb2c(261)]);
    }
    this["userData"] = { "id": _0x3910ac["id"], "is_premium": _0x3910ac[_0x23eb2c(249)], "first_name": _0x3910ac[_0x23eb2c(262)], "is_bot": _0x3910ac[_0x23eb2c(230)], "last_name": _0x3910ac[_0x23eb2c(250)], "language_code": _0x3910ac[_0x23eb2c(248)], "photo_url": _0x3910ac["photoUrl"], "username": _0x3910ac[_0x23eb2c(240)] }, this[_0x23eb2c(234)] = _0x3910ac["id"], this[_0x23eb2c(245)] = _0x3910ac["languageCode"], this[_0x23eb2c(227)] = _0x24cdcd[_0x23eb2c(247)], this["platform"] = _0x258462[_0x23eb2c(266)], this[_0x23eb2c(244)] = generateUUID(_0x408df5[_0x23eb2c(235)](String, this["getUserId"]()));
  }
  ["getSessionId"]() {
    const _0x1fdf47 = _0x34f128;
    return this[_0x1fdf47(244)];
  }
  [_0x34f128(269)]() {
    const _0x3a3f0b = _0x34f128;
    return this[_0x3a3f0b(234)];
  }
  [_0x34f128(243)]() {
    const _0x592c8f = _0x34f128;
    return this[_0x592c8f(227)];
  }
  ["getPlatform"]() {
    const _0x38d155 = _0x34f128;
    return this[_0x38d155(266)];
  }
  [_0x34f128(268)]() {
    const _0x4f40d6 = _0x34f128;
    return this[_0x4f40d6(245)];
  }
  [_0x34f128(255)]() {
    const _0x28fb10 = _0x34f128;
    return this[_0x28fb10(251)];
  }
  [_0x34f128(238)]() {
    const _0x5b96c2 = _0x34f128, _0x57a381 = { "fiFAH": function(_0x2d961e, _0x8a3bb8) {
      return _0x2d961e(_0x8a3bb8);
    } }, _0x47a2a9 = this[_0x5b96c2(255)]();
    return _0x57a381[_0x5b96c2(260)](Boolean, _0x47a2a9 == null ? void 0 : _0x47a2a9[_0x5b96c2(254)]);
  }
  [_0x34f128(267)]() {
    const _0x22a70c = _0x34f128, _0x2302f7 = { "vpqwL": function(_0xd0f0aa, _0x5bc5c6) {
      return _0xd0f0aa(_0x5bc5c6);
    } };
    return { "session_id": this["getSessionId"](), "user_id": this[_0x22a70c(269)](), "app_name": this["appModule"][_0x22a70c(256)](), "is_premium": this["getUserIsPremium"](), "platform": this[_0x22a70c(239)](), "locale": this[_0x22a70c(268)](), "start_param": this[_0x22a70c(243)](), "client_timestamp": _0x2302f7[_0x22a70c(265)](String, Date[_0x22a70c(228)]()) };
  }
}
const _0x10ef38 = _0x282d;
function _0x282d(_0x32cbca, _0x4f064c) {
  const _0x1b3013 = _0x1b30();
  return _0x282d = function(_0x282da3, _0x7c1dc5) {
    _0x282da3 = _0x282da3 - 492;
    let _0x8c8545 = _0x1b3013[_0x282da3];
    return _0x8c8545;
  }, _0x282d(_0x32cbca, _0x4f064c);
}
(function(_0x3dbad3, _0x5a3c63) {
  const _0x15f880 = _0x282d, _0x2a4950 = _0x3dbad3();
  while (!![]) {
    try {
      const _0x3b34bf = parseInt(_0x15f880(500)) / 1 + -parseInt(_0x15f880(509)) / 2 * (-parseInt(_0x15f880(496)) / 3) + -parseInt(_0x15f880(518)) / 4 + parseInt(_0x15f880(494)) / 5 + -parseInt(_0x15f880(502)) / 6 + parseInt(_0x15f880(493)) / 7 * (-parseInt(_0x15f880(505)) / 8) + parseInt(_0x15f880(492)) / 9;
      if (_0x3b34bf === _0x5a3c63) break;
      else _0x2a4950["push"](_0x2a4950["shift"]());
    } catch (_0x28d1f9) {
      _0x2a4950["push"](_0x2a4950["shift"]());
    }
  }
})(_0x1b30, 142007);
function _0x1b30() {
  const _0x35aeda = ["hZSIW", "stringify", "25372vFeGbp", "localStorage", "FgEjc", "push", "sessionStorage", "null", "getBatch", "parse", "findIndex", "754348NeyRdE", "2073150xUnvHQ", "283773tBGYeG", "445465vESSbn", "setItem", "36rQFsyo", "HZLZv", "key", "UHVBo", "188307GImAaJ", "getItem", "516924juLfPK", "includes", "xngnR", "48XUGuli", "qlxbw"];
  _0x1b30 = function() {
    return _0x35aeda;
  };
  return _0x1b30();
}
class BatchStorage {
  constructor(_0x1dec59) {
    const _0x2f7a74 = _0x282d;
    this["sessionStorage"] = window[_0x2f7a74(513)], this[_0x2f7a74(510)] = window[_0x2f7a74(510)], this[_0x2f7a74(498)] = _0x1dec59;
  }
  ["getBatch"]() {
    const _0xe51b45 = _0x282d, _0x59d481 = { "FgEjc": _0xe51b45(514), "mQsth": function(_0x902c02, _0x1a9db2) {
      return _0x902c02 !== _0x1a9db2;
    }, "qlxbw": "hZSIW", "xngnR": function(_0x32fd0b, _0x8bf7d7) {
      return _0x32fd0b !== _0x8bf7d7;
    }, "HZLZv": _0xe51b45(499) };
    return [null, "null"][_0xe51b45(503)](this[_0xe51b45(513)][_0xe51b45(501)](this["key"])) && [null, _0x59d481[_0xe51b45(511)]][_0xe51b45(503)](this[_0xe51b45(510)][_0xe51b45(501)](this[_0xe51b45(498)])) ? _0x59d481["mQsth"](_0x59d481[_0xe51b45(506)], _0xe51b45(507)) ? (this["sessionStorage"] = _0xf2e9b9[_0xe51b45(513)], this["localStorage"] = _0x4ceea6[_0xe51b45(510)], this["key"] = _0x2e1c5a) : this[_0xe51b45(495)]([]) : _0x59d481[_0xe51b45(504)](_0x59d481[_0xe51b45(497)], _0x59d481["HZLZv"]) ? this[_0xe51b45(495)](_0x40204a[_0xe51b45(516)](this["localStorage"][_0xe51b45(501)](this[_0xe51b45(498)]))) : this[_0xe51b45(495)](JSON[_0xe51b45(516)](this[_0xe51b45(510)][_0xe51b45(501)](this["key"]))), this[_0xe51b45(495)]([...JSON[_0xe51b45(516)](this[_0xe51b45(513)]["getItem"](this["key"])), ...JSON[_0xe51b45(516)](this[_0xe51b45(510)][_0xe51b45(501)](this["key"]))]["filter"]((_0x29a2c7, _0x2791b3, _0x2a81cf) => _0x2a81cf[_0xe51b45(517)]((_0x4eaf2b) => JSON["stringify"](_0x4eaf2b) === JSON[_0xe51b45(508)](_0x29a2c7)) === _0x2791b3)), JSON[_0xe51b45(516)](this[_0xe51b45(513)][_0xe51b45(501)](this[_0xe51b45(498)]));
  }
  ["addToStorage"](_0x4d5255, _0x4a6a5e) {
    const _0x4a746d = _0x282d, _0x36f1a0 = this[_0x4a746d(515)]();
    _0x36f1a0[_0x4a746d(512)]({ "event_name": _0x4d5255, ..._0x4a6a5e }), this["setItem"](_0x36f1a0);
  }
  [_0x10ef38(495)](_0x22dadf) {
    const _0x3c8b2e = _0x10ef38;
    this[_0x3c8b2e(510)][_0x3c8b2e(495)](this["key"], JSON[_0x3c8b2e(508)](_0x22dadf)), this["sessionStorage"][_0x3c8b2e(495)](this[_0x3c8b2e(498)], JSON["stringify"](_0x22dadf));
  }
}
function _0x3949(_0x3225d0, _0x156626) {
  const _0x3a71f2 = _0x3a71();
  return _0x3949 = function(_0x39499f, _0x1d71df) {
    _0x39499f = _0x39499f - 249;
    let _0x1cec33 = _0x3a71f2[_0x39499f];
    return _0x1cec33;
  }, _0x3949(_0x3225d0, _0x156626);
}
const _0x5c90f4 = _0x3949;
(function(_0x1a4f0f, _0x872d5d) {
  const _0x270862 = _0x3949, _0x3af352 = _0x1a4f0f();
  while (!![]) {
    try {
      const _0x4afa00 = parseInt(_0x270862(297)) / 1 * (-parseInt(_0x270862(366)) / 2) + -parseInt(_0x270862(324)) / 3 + parseInt(_0x270862(342)) / 4 + parseInt(_0x270862(350)) / 5 * (parseInt(_0x270862(254)) / 6) + -parseInt(_0x270862(271)) / 7 + parseInt(_0x270862(269)) / 8 * (parseInt(_0x270862(249)) / 9) + -parseInt(_0x270862(320)) / 10 * (-parseInt(_0x270862(327)) / 11);
      if (_0x4afa00 === _0x872d5d) break;
      else _0x3af352["push"](_0x3af352["shift"]());
    } catch (_0x7f3ce9) {
      _0x3af352["push"](_0x3af352["shift"]());
    }
  }
})(_0x3a71, 175232);
function _0x3a71() {
  const _0x56d8cc = ["status", "IUoOP", "eYEQo", "MBIzm", "AXHtu", "vyjmf", "qCqZm", "processQueue", "QuhXF", "qdDIK", "appModule", "FveiA", "hxFsg", "SvlZO", "startBatchingWithInterval", "onreadystatechange", "some", "ttQYE", "197aOyNHx", "length", "addToStorage", "uqHux", "VXFnR", "rdByU", "addEventListener", "iznIm", "collectEvent", "owWmX", "taskRetry", "ijtaj", "rFloJ", "hBXhG", "init", "DHLvO", "PnlEL", "stopBatching", "rnlRK", "EMVzF", "kbGar", "recordEvents", "getApiToken", "355620WJTxos", "YmeXW", "setInterval", "euceK", "776412KBviFI", "VhNcP", "mlmTd", "77RMuKHd", "RgMuu", "fCERi", "FPTZD", "mWKTo", "shNOB", "xdWWK", "aXapY", "taskSolution", "temot", "uWxkm", "WEQCe", "storage", "Igfev", "BATCH_KEY", "1063824lWeAAe", "qPDSN", "BVNfI", "429", "sendBatch", "GpLos", "getBatch", "DXJsH", "5DAAaee", "startBatching", "zUaOy", "SVejh", "fdTkX", "KaHfi", "yOiie", "DzhXT", "ORmzN", "log", "DUIwF", "load", "solveTask", "readyState", "then", "BNcrq", "1004jMnxDG", "3609zVBmhk", "onLine", "UvFyD", "backoff", "batchInterval", "1518612nlnJfG", "xIrpH", "yQKob", "zwNuA", "tUdkr", "navigator", "UjYNv", "EkvfP", "slice", "complete", "stringify", "JVZJP", "filter", "FJAjp", "intervalId", "248XZoYbF", "collect", "1732451XpbwaH", "ZgLGx", "SnSKA", "AqaFb", "AKBfV", "BsRVK", "kRtMo", "YeZiP"];
  _0x3a71 = function() {
    return _0x56d8cc;
  };
  return _0x3a71();
}
class BatchService {
  constructor(_0x5c11d2) {
    const _0x53f907 = _0x3949, _0x113d71 = { "FJAjp": function(_0x97023b, _0x5c9607) {
      return _0x97023b + _0x5c9607;
    } };
    this["backoff"] = 1, this["intervalId"] = null, this[_0x53f907(253)] = 2e3, this[_0x53f907(307)] = 0, this[_0x53f907(341)] = BATCH_KEY, this[_0x53f907(289)] = _0x5c11d2, this[_0x53f907(339)] = new BatchStorage(_0x113d71[_0x53f907(267)](_0x113d71["FJAjp"](this["BATCH_KEY"], "-"), this[_0x53f907(289)][_0x53f907(319)]()));
  }
  [_0x5c90f4(311)]() {
    const _0x5ad19b = _0x5c90f4, _0x8b8746 = { "GpLos": function(_0x576966, _0x40929a) {
      return _0x576966 >= _0x40929a;
    }, "PnlEL": function(_0xc53fe1, _0x1d6fa3) {
      return _0xc53fe1 == _0x1d6fa3;
    }, "UjYNv": function(_0x4b592b, _0x167811) {
      return _0x4b592b === _0x167811;
    }, "yzYhb": _0x5ad19b(332), "ijtaj": "QjftB", "AKBfV": _0x5ad19b(263) };
    _0x8b8746[_0x5ad19b(260)](document[_0x5ad19b(363)], _0x8b8746[_0x5ad19b(275)]) ? this["startBatchingWithInterval"]() : document[_0x5ad19b(294)] = () => {
      const _0x41e77c = _0x5ad19b, _0x30932e = { "IUoOP": function(_0x50209d, _0x227ff5) {
        const _0x439705 = _0x3949;
        return _0x8b8746[_0x439705(347)](_0x50209d, _0x227ff5);
      } };
      if (_0x8b8746[_0x41e77c(313)](document[_0x41e77c(363)], _0x41e77c(263))) {
        if (_0x8b8746["UjYNv"](_0x8b8746["yzYhb"], _0x8b8746[_0x41e77c(308)])) {
          if (_0x30932e[_0x41e77c(280)](_0x4367c8++, 3)) {
            this["startBatching"](), _0x1a7b35(_0x2182b4);
            return;
          }
          this[_0x41e77c(289)][_0x41e77c(362)]();
        } else this[_0x41e77c(293)]();
      }
    };
  }
  [_0x5c90f4(293)]() {
    const _0x47331a = _0x5c90f4, _0xafae95 = { "hBXhG": function(_0x14fc88, _0xada3d8) {
      return _0x14fc88 !== _0xada3d8;
    }, "AqaFb": function(_0x486534, _0x165797) {
      return _0x486534(_0x165797);
    }, "xIrpH": function(_0xbeac76, _0x182363) {
      return _0xbeac76 * _0x182363;
    }, "SJnWA": _0x47331a(281), "ZgLGx": _0x47331a(282), "xdWWK": function(_0x47a6d2, _0x13ea1c) {
      return _0x47a6d2 !== _0x13ea1c;
    }, "zUaOy": _0x47331a(334), "DHLvO": _0x47331a(302), "DUIwF": function(_0x671404, _0x203b11) {
      return _0x671404 === _0x203b11;
    }, "euceK": _0x47331a(285), "VXFnR": "EULNh", "VhNcP": function(_0x413200, _0x12e2ae) {
      return _0x413200 >= _0x12e2ae;
    }, "yQKob": function(_0x171748, _0x153e23) {
      return _0x171748(_0x153e23);
    }, "FveiA": function(_0x1feceb, _0x361afe, _0x870ffb) {
      return _0x1feceb(_0x361afe, _0x870ffb);
    } };
    let _0x14bc5e = 0;
    this[_0x47331a(289)]["solveTask"](), this[_0x47331a(289)][_0x47331a(305)](Events["INIT"]);
    if (_0xafae95[_0x47331a(310)](this[_0x47331a(289)]["taskSolution"], void 0)) this[_0x47331a(351)]();
    else {
      const _0x50a2c2 = _0xafae95[_0x47331a(290)](setInterval, () => {
        const _0x264382 = _0x47331a, _0x1e37a2 = { "UvFyD": function(_0x1f41d7, _0x16ed00) {
          const _0x1616b3 = _0x3949;
          return _0xafae95[_0x1616b3(255)](_0x1f41d7, _0x16ed00);
        } };
        if (_0xafae95["SJnWA"] !== _0xafae95[_0x264382(272)]) {
          if (this[_0x264382(289)][_0x264382(335)] !== void 0) _0xafae95[_0x264382(333)](_0xafae95[_0x264382(352)], _0xafae95[_0x264382(312)]) ? (this[_0x264382(351)](), _0xafae95[_0x264382(274)](clearInterval, _0x50a2c2)) : (this[_0x264382(252)]++, this[_0x264382(253)] = _0x1e37a2[_0x264382(251)](this["batchInterval"], 2.71), this["startBatching"]());
          else {
            if (_0xafae95[_0x264382(360)](_0xafae95[_0x264382(323)], _0xafae95[_0x264382(301)])) _0xafae95[_0x264382(310)](this["intervalId"], null) && (_0xafae95[_0x264382(274)](_0x112ab1, this[_0x264382(268)]), this[_0x264382(268)] = null);
            else {
              if (_0xafae95[_0x264382(325)](_0x14bc5e++, 3)) {
                this[_0x264382(351)](), _0xafae95[_0x264382(256)](clearInterval, _0x50a2c2);
                return;
              }
              this[_0x264382(289)]["solveTask"]();
            }
          }
        } else return;
      }, 1e3);
    }
  }
  [_0x5c90f4(314)]() {
    const _0x3181f4 = _0x5c90f4, _0x1202d1 = { "BNcrq": function(_0x1f58a5, _0x55e1c0) {
      return _0x1f58a5 !== _0x55e1c0;
    }, "fCERi": function(_0x1bc628, _0x8770c) {
      return _0x1bc628(_0x8770c);
    } };
    _0x1202d1[_0x3181f4(365)](this[_0x3181f4(268)], null) && (_0x1202d1[_0x3181f4(329)](clearInterval, this[_0x3181f4(268)]), this[_0x3181f4(268)] = null);
  }
  [_0x5c90f4(270)](_0x3eedca, _0x9f68a0) {
    const _0x2cb5e7 = _0x5c90f4, _0x59a229 = { "hxFsg": function(_0x1c7c67, _0x179def) {
      return _0x1c7c67 !== _0x179def;
    }, "zwNuA": function(_0x4e139e, _0x159d6a) {
      return _0x4e139e === _0x159d6a;
    }, "BVNfI": _0x2cb5e7(277), "mlmTd": _0x2cb5e7(263), "YeZiP": function(_0x2c02e4, _0x793b7) {
      return _0x2c02e4 !== _0x793b7;
    }, "YmeXW": _0x2cb5e7(331), "SnSKA": function(_0x4a3b92) {
      return _0x4a3b92();
    }, "JVZJP": _0x2cb5e7(340), "QuhXF": _0x2cb5e7(361), "yOiie": function(_0x3dd829, _0x1aa84f, _0x2b7170) {
      return _0x3dd829(_0x1aa84f, _0x2b7170);
    } };
    if (_0x59a229[_0x2cb5e7(257)](document[_0x2cb5e7(363)], _0x59a229[_0x2cb5e7(326)])) this[_0x2cb5e7(339)][_0x2cb5e7(299)](_0x3eedca, _0x9f68a0);
    else {
      const _0x7be846 = () => {
        const _0x93c135 = _0x2cb5e7;
        this[_0x93c135(339)]["addToStorage"](_0x3eedca, _0x9f68a0);
      };
      _0x59a229[_0x2cb5e7(356)](setTimeout, () => {
        const _0x5b975d = _0x2cb5e7, _0x5a6cbd = { "tUdkr": function(_0x5c7f9a, _0x42a6fa) {
          const _0x4cf356 = _0x3949;
          return _0x59a229[_0x4cf356(291)](_0x5c7f9a, _0x42a6fa);
        } };
        if (_0x59a229[_0x5b975d(257)](_0x59a229[_0x5b975d(344)], _0x59a229["BVNfI"])) {
          if (_0x59a229[_0x5b975d(257)](document[_0x5b975d(363)], _0x59a229[_0x5b975d(326)])) _0x59a229[_0x5b975d(278)](_0x59a229[_0x5b975d(321)], _0x59a229["YmeXW"]) ? this["storage"]["addToStorage"](_0x385a21, _0x425f2b) : _0x59a229[_0x5b975d(273)](_0x7be846);
          else {
            if (_0x59a229["zwNuA"](_0x59a229[_0x5b975d(265)], "vmemm")) {
              const _0x3597ef = this[_0x5b975d(339)][_0x5b975d(348)]();
              _0x5a6cbd[_0x5b975d(258)](_0x3597ef["length"], 0) && _0x48dfbb[_0x5b975d(259)][_0x5b975d(250)] && this[_0x5b975d(346)](_0x3597ef["slice"](0, 20));
            } else window[_0x5b975d(303)](_0x59a229[_0x5b975d(287)], _0x7be846);
          }
        } else this["startBatching"](), _0x382387(_0x1a3595);
      }, 0);
    }
  }
  [_0x5c90f4(351)]() {
    const _0x394805 = _0x5c90f4, _0x22146d = { "RgMuu": function(_0x2872b0, _0x390037) {
      return _0x2872b0 === _0x390037;
    } };
    this[_0x394805(289)][_0x394805(362)](), _0x22146d[_0x394805(328)](this["intervalId"], null) && (this[_0x394805(268)] = window[_0x394805(322)](() => this[_0x394805(286)](), this["batchInterval"]));
  }
  [_0x5c90f4(286)]() {
    const _0x286caf = _0x5c90f4, _0x1887b5 = { "uqHux": function(_0x489965, _0x4bce52) {
      return _0x489965(_0x4bce52);
    }, "fdTkX": function(_0x255dd7, _0x4459a6) {
      return _0x255dd7 !== _0x4459a6;
    }, "AXHtu": function(_0x20a181, _0x4ec355) {
      return _0x20a181 === _0x4ec355;
    }, "qdDIK": _0x286caf(316), "Isgpj": _0x286caf(349) }, _0x12efe0 = this[_0x286caf(339)]["getBatch"]();
    if (_0x1887b5[_0x286caf(354)](_0x12efe0[_0x286caf(298)], 0) && window[_0x286caf(259)][_0x286caf(250)]) {
      if (_0x1887b5[_0x286caf(283)](_0x1887b5[_0x286caf(288)], _0x1887b5["Isgpj"])) {
        this[_0x286caf(351)](), _0x1887b5[_0x286caf(300)](_0x568339, _0x4cf6ae);
        return;
      } else this[_0x286caf(346)](_0x12efe0[_0x286caf(262)](0, 20));
    }
  }
  ["sendBatch"](_0x2c8d62) {
    const _0x4159a8 = _0x5c90f4, _0x37dd9a = { "WVpzU": function(_0x1ea920, _0x2f657b) {
      return _0x1ea920 == _0x2f657b;
    }, "EkvfP": function(_0x21cdfa, _0x246784) {
      return _0x21cdfa === _0x246784;
    }, "xmvvv": function(_0x5246de, _0x27d109) {
      return _0x5246de(_0x27d109);
    }, "KaHfi": _0x4159a8(358), "ZtYnW": function(_0x239842, _0x455781) {
      return _0x239842 === _0x455781;
    }, "WEQCe": function(_0x4e3e41, _0x1f16d6) {
      return _0x4e3e41 === _0x1f16d6;
    }, "ttQYE": function(_0x5301d3, _0x20f1de) {
      return _0x5301d3(_0x20f1de);
    }, "ModsC": function(_0x1d34c3, _0x5a05f9) {
      return _0x1d34c3 < _0x5a05f9;
    }, "qfDhl": function(_0x333f4d, _0x3248f7) {
      return _0x333f4d * _0x3248f7;
    }, "kbGar": function(_0x1c12bd, _0x344963) {
      return _0x1c12bd !== _0x344963;
    }, "temot": "IviMN", "FPTZD": function(_0x5c4c08, _0x5a7190) {
      return _0x5c4c08 > _0x5a7190;
    }, "SVejh": function(_0x49595a, _0x346a24) {
      return _0x49595a !== _0x346a24;
    }, "rnlRK": _0x4159a8(304), "sxCiX": _0x4159a8(284), "qPDSN": function(_0x257e43, _0x501d3a) {
      return _0x257e43 !== _0x501d3a;
    }, "owWmX": function(_0xf701b9, _0x18a107) {
      return _0xf701b9(_0x18a107);
    }, "SvlZO": function(_0x3dfbfa, _0x3d559d) {
      return _0x3dfbfa >= _0x3d559d;
    }, "DzhXT": function(_0x1c02bb, _0x2e17fc) {
      return _0x1c02bb !== _0x2e17fc;
    } };
    this[_0x4159a8(314)](), this["appModule"][_0x4159a8(318)](_0x2c8d62)[_0x4159a8(364)]((_0x297298) => {
      const _0x26ba5d = _0x4159a8, _0x1a350c = { "MWLbn": function(_0xe5ab30, _0x4d2f95) {
        return _0x37dd9a["WVpzU"](_0xe5ab30, _0x4d2f95);
      }, "uWxkm": function(_0x290247, _0xeec47) {
        return _0x290247(_0xeec47);
      } };
      if (_0x37dd9a[_0x26ba5d(261)](_0x37dd9a["xmvvv"](String, _0x297298[_0x26ba5d(279)]), _0x26ba5d(345))) {
        if (_0x37dd9a[_0x26ba5d(261)](_0x37dd9a[_0x26ba5d(355)], _0x26ba5d(358))) {
          this["startBatching"]();
          return;
        } else _0x1a350c["MWLbn"](_0x32521d[_0x26ba5d(363)], _0x26ba5d(263)) && this[_0x26ba5d(293)]();
      }
      if (_0x37dd9a["ZtYnW"](String(_0x297298[_0x26ba5d(279)])[0], "4")) return;
      if (_0x37dd9a[_0x26ba5d(338)](_0x37dd9a[_0x26ba5d(296)](String, _0x297298[_0x26ba5d(279)])[0], "5")) {
        _0x37dd9a["ModsC"](this["backoff"], 5) && (this[_0x26ba5d(252)]++, this[_0x26ba5d(253)] = _0x37dd9a["qfDhl"](this[_0x26ba5d(253)], 2.71), this[_0x26ba5d(351)]());
        return;
      }
      this[_0x26ba5d(252)] = 1, this[_0x26ba5d(253)] = 2e3;
      if (_0x37dd9a[_0x26ba5d(338)](String(_0x297298[_0x26ba5d(279)]), "203")) {
        if (_0x37dd9a[_0x26ba5d(317)](_0x37dd9a[_0x26ba5d(336)], _0x26ba5d(309))) {
          this[_0x26ba5d(307)]++, this["appModule"][_0x26ba5d(335)] = void 0;
          if (_0x37dd9a[_0x26ba5d(330)](this[_0x26ba5d(307)], 3)) {
            if (_0x37dd9a[_0x26ba5d(353)](_0x37dd9a[_0x26ba5d(315)], _0x37dd9a["sxCiX"])) {
              this["startBatching"]();
              return;
            } else this[_0x26ba5d(293)]();
          } else {
            this[_0x26ba5d(289)][_0x26ba5d(362)](), this[_0x26ba5d(351)]();
            return;
          }
        } else _0x1a350c[_0x26ba5d(337)](_0x75c6db, this[_0x26ba5d(268)]), this[_0x26ba5d(268)] = null;
      }
      this[_0x26ba5d(307)] = 0, this[_0x26ba5d(339)]["setItem"](this["storage"][_0x26ba5d(348)]()[_0x26ba5d(266)]((_0x4db4b0) => !_0x2c8d62[_0x26ba5d(295)]((_0x2fec09) => JSON[_0x26ba5d(264)](_0x4db4b0) === JSON[_0x26ba5d(264)](_0x2fec09)))), this[_0x26ba5d(351)]();
    }, (_0x317cac) => {
      const _0x3be138 = _0x4159a8;
      if (_0x37dd9a[_0x3be138(357)]("RBJBF", _0x3be138(276))) console[_0x3be138(359)](_0x317cac), this[_0x3be138(351)]();
      else {
        if (_0x37dd9a[_0x3be138(343)](this[_0x3be138(289)][_0x3be138(335)], void 0)) this[_0x3be138(351)](), _0x37dd9a[_0x3be138(306)](_0x320cb8, _0x10d79f);
        else {
          if (_0x37dd9a[_0x3be138(292)](_0x4c3387++, 3)) {
            this[_0x3be138(351)](), _0x4398a1(_0x2cf4dc);
            return;
          }
          this[_0x3be138(289)][_0x3be138(362)]();
        }
      }
    });
  }
}
var _0x3e2697 = _0x429d;
(function(_0x561535, _0x4f74de) {
  var _0x4f7c18 = _0x429d, _0xfe813c = _0x561535();
  while (!![]) {
    try {
      var _0x5220e4 = -parseInt(_0x4f7c18(226)) / 1 * (parseInt(_0x4f7c18(244)) / 2) + parseInt(_0x4f7c18(246)) / 3 + -parseInt(_0x4f7c18(254)) / 4 + -parseInt(_0x4f7c18(233)) / 5 * (parseInt(_0x4f7c18(241)) / 6) + parseInt(_0x4f7c18(223)) / 7 * (-parseInt(_0x4f7c18(242)) / 8) + -parseInt(_0x4f7c18(237)) / 9 * (-parseInt(_0x4f7c18(249)) / 10) + parseInt(_0x4f7c18(252)) / 11;
      if (_0x5220e4 === _0x4f74de) break;
      else _0xfe813c["push"](_0xfe813c["shift"]());
    } catch (_0x50d613) {
      _0xfe813c["push"](_0xfe813c["shift"]());
    }
  }
})(_0x59ef, 654751);
class HumanProofService {
  constructor(_0x26066d) {
    var _0x4a6e47 = _0x429d;
    this[_0x4a6e47(224)] = _0x26066d;
  }
  async ["init"]() {
    var _0x5c0a30 = _0x429d, _0xb076b1 = { "aWQIN": function(_0x1d2347, _0x325108) {
      return _0x1d2347(_0x325108);
    }, "jqKHi": _0x5c0a30(235), "slnrX": function(_0x47b0b5) {
      return _0x47b0b5();
    }, "ZjQxL": function(_0x59de5f, _0x21ec78) {
      return _0x59de5f(_0x21ec78);
    }, "UbFho": function(_0x4c2b99, _0x25efdc) {
      return _0x4c2b99 + _0x25efdc;
    }, "QOaEz": _0x5c0a30(240) };
    return new Promise(async (_0x40139a, _0x3b908c) => {
      var _0x51c75e = _0x5c0a30, _0x2fd06e = { "EEPao": function(_0x5cda85, _0x38b215) {
        var _0x39aac8 = _0x429d;
        return _0xb076b1[_0x39aac8(251)](_0x5cda85, _0x38b215);
      }, "FgyBA": function(_0x5ddd38) {
        var _0x1c678c = _0x429d;
        return _0xb076b1[_0x1c678c(229)](_0x5ddd38);
      } };
      try {
        await _0xb076b1[_0x51c75e(230)](fetch, _0xb076b1[_0x51c75e(234)](BACKEND_URL, _0xb076b1[_0x51c75e(227)]))[_0x51c75e(250)](async (_0x550c38) => {
          var _0x432950 = _0x51c75e;
          this[_0x432950(232)] = new Worker(URL[_0x432950(231)](await _0x550c38[_0x432950(239)]())), await _0xb076b1[_0x432950(251)](fetch, BACKEND_URL + _0xb076b1[_0x432950(247)])[_0x432950(250)](async (_0xa97a02) => {
            var _0x1f2a90 = _0x432950;
            this[_0x1f2a90(224)][_0x1f2a90(248)] = _0x2fd06e[_0x1f2a90(245)](String, await _0xa97a02["text"]()), this[_0x1f2a90(232)][_0x1f2a90(225)] = (_0x33ab5d) => {
              var _0x49088a = _0x1f2a90;
              this["appModule"][_0x49088a(236)] = _0x33ab5d[_0x49088a(228)];
            }, _0x2fd06e["FgyBA"](_0x40139a);
          });
        });
      } catch (_0x2a557d) {
        _0x3b908c(_0x2a557d);
      }
    });
  }
  [_0x3e2697(238)](_0x3376e4) {
    var _0x4189e8 = _0x3e2697;
    this[_0x4189e8(224)][_0x4189e8(248)] = _0x3376e4, this[_0x4189e8(243)]();
  }
  [_0x3e2697(243)]() {
    var _0x5a1ca5 = _0x3e2697, _0x112fc4 = { "GoZAi": function(_0x587d14, _0x2415a4) {
      return _0x587d14 !== _0x2415a4;
    } };
    _0x112fc4[_0x5a1ca5(253)](this[_0x5a1ca5(232)], void 0) && this[_0x5a1ca5(232)]["postMessage"](this["appModule"][_0x5a1ca5(248)]);
  }
}
function _0x429d(_0x57cb6a, _0x5859f7) {
  var _0x59eff4 = _0x59ef();
  return _0x429d = function(_0x429ded, _0xdc2a6c) {
    _0x429ded = _0x429ded - 223;
    var _0x189944 = _0x59eff4[_0x429ded];
    return _0x189944;
  }, _0x429d(_0x57cb6a, _0x5859f7);
}
function _0x59ef() {
  var _0x3f7fe3 = ["onmessage", "313838zRltDT", "QOaEz", "data", "slnrX", "ZjQxL", "createObjectURL", "worker", "53765zTpdXg", "UbFho", "aee7c93a9ae7930fb19732325d2c560c53849aa7", "taskSolution", "108MMUMBl", "setNewArgs", "blob", "c3e068ebf11840ed3fc311a6f2df80b20fa05d25", "42JuPQhh", "1816dSfOJt", "solveTask", "4KLTxUG", "EEPao", "1787082noOaSw", "jqKHi", "taskParams", "64510INJXXX", "then", "aWQIN", "22212586kwmiCC", "GoZAi", "1992956nFrWDX", "25795FdJqzs", "appModule"];
  _0x59ef = function() {
    return _0x3f7fe3;
  };
  return _0x59ef();
}
var _0x596997 = _0x49be;
(function(_0x1c5810, _0x4ef199) {
  var _0x34fa66 = _0x49be, _0x38c286 = _0x1c5810();
  while (!![]) {
    try {
      var _0x300c04 = parseInt(_0x34fa66(261)) / 1 + parseInt(_0x34fa66(247)) / 2 + -parseInt(_0x34fa66(250)) / 3 + parseInt(_0x34fa66(268)) / 4 + -parseInt(_0x34fa66(248)) / 5 * (-parseInt(_0x34fa66(275)) / 6) + -parseInt(_0x34fa66(260)) / 7 + parseInt(_0x34fa66(254)) / 8 * (-parseInt(_0x34fa66(262)) / 9);
      if (_0x300c04 === _0x4ef199) break;
      else _0x38c286["push"](_0x38c286["shift"]());
    } catch (_0xb0a3ea) {
      _0x38c286["push"](_0x38c286["shift"]());
    }
  }
})(_0x1cdb, 587790);
function _0x49be(_0x98fd23, _0x523815) {
  var _0x1cdb72 = _0x1cdb();
  return _0x49be = function(_0x49be0c, _0x8d2c42) {
    _0x49be0c = _0x49be0c - 247;
    var _0x51c7aa = _0x1cdb72[_0x49be0c];
    return _0x51c7aa;
  }, _0x49be(_0x98fd23, _0x523815);
}
function _0x1cdb() {
  var _0x682733 = ["0|1|4|2|3", "4038056DTafAu", "collectEvent", "split", "5|4|6|2|0|1|3|7", "recordEvents", "getUserData", "assembleEventSession", "383322zcTwLp", "getAppName", "collect", "recordEvent", "324742VmyKNG", "25xEroqP", "collectTappsEvent", "634917Wkbwkd", "networkController", "analyticsController", "humanProofService", "58120UbsOIm", "catch", "setNewArgs", "init", "appName", "apiToken", "8185646arpiYg", "1007832CBAklt", "657FFEyjJ", "batchService", "sessionController", "solveTask", "env"];
  _0x1cdb = function() {
    return _0x682733;
  };
  return _0x1cdb();
}
class App {
  constructor(_0x2d789c, _0x1c8c6c, _0x4f4480) {
    var _0x52bfca = _0x49be, _0x3ac6d7 = _0x52bfca(271)["split"]("|"), _0x33dcd9 = 0;
    while (!![]) {
      switch (_0x3ac6d7[_0x33dcd9++]) {
        case "0":
          this["sessionController"] = new SessionController(this);
          continue;
        case "1":
          this[_0x52bfca(251)] = new NetworkController(this);
          continue;
        case "2":
          this[_0x52bfca(253)] = new HumanProofService(this);
          continue;
        case "3":
          this[_0x52bfca(252)] = new AnalyticsController(this);
          continue;
        case "4":
          this[_0x52bfca(259)] = _0x2d789c;
          continue;
        case "5":
          this[_0x52bfca(266)] = _0x4f4480;
          continue;
        case "6":
          this[_0x52bfca(258)] = _0x1c8c6c;
          continue;
        case "7":
          this[_0x52bfca(263)] = new BatchService(this);
          continue;
      }
      break;
    }
  }
  async [_0x596997(257)]() {
    var _0x59e060 = _0x596997, _0x8caeea = _0x59e060(267)[_0x59e060(270)]("|"), _0x7ecf1c = 0;
    while (!![]) {
      switch (_0x8caeea[_0x7ecf1c++]) {
        case "0":
          this["sessionController"][_0x59e060(257)]();
          continue;
        case "1":
          await this[_0x59e060(252)][_0x59e060(257)]();
          continue;
        case "2":
          this["networkController"][_0x59e060(257)]();
          continue;
        case "3":
          this[_0x59e060(263)][_0x59e060(257)]();
          continue;
        case "4":
          await this["humanProofService"]["init"]()["then"](() => {
            var _0x2dff42 = _0x59e060;
            this[_0x2dff42(265)]();
          })[_0x59e060(255)]((_0x436bbf) => console["error"](_0x436bbf));
          continue;
      }
      break;
    }
  }
  ["assembleEventSession"]() {
    var _0x1af5ec = _0x596997;
    return this[_0x1af5ec(264)]["assembleEventSession"]();
  }
  [_0x596997(278)](_0x3fbcb3, _0x5ca76e, _0x3dedf8) {
    var _0x1ba07d = _0x596997;
    return this[_0x1ba07d(251)][_0x1ba07d(278)](_0x3fbcb3, _0x5ca76e, _0x3dedf8);
  }
  [_0x596997(272)](_0x538021) {
    var _0xeb6af8 = _0x596997;
    return this[_0xeb6af8(251)][_0xeb6af8(272)](_0x538021);
  }
  [_0x596997(269)](_0x20fdbd, _0x148635) {
    var _0x2ab89e = _0x596997;
    this[_0x2ab89e(263)][_0x2ab89e(277)](_0x20fdbd, { ..._0x148635, ...this[_0x2ab89e(274)]() });
  }
  ["registerInvoice"](_0x47dbbe) {
    var _0x41088f = _0x596997;
    this["batchService"][_0x41088f(277)](Events["INVOICE_REGISTERED"], { ..._0x47dbbe, ...this["assembleEventSession"]() });
  }
  [_0x596997(249)](_0x1fa20c, _0x2eb3a9) {
    var _0x37ff78 = _0x596997;
    this[_0x37ff78(263)]["collect"](_0x1fa20c, { ...this[_0x37ff78(264)][_0x37ff78(274)](), "custom_data": { "userData": { ...this[_0x37ff78(264)][_0x37ff78(273)]() }, ..._0x2eb3a9 } });
  }
  ["getApiToken"]() {
    var _0xd3d32 = _0x596997;
    return this[_0xd3d32(259)];
  }
  [_0x596997(276)]() {
    var _0x1dad6e = _0x596997;
    return this[_0x1dad6e(258)];
  }
  ["solveTask"]() {
    var _0x2f3256 = _0x596997;
    this[_0x2f3256(253)][_0x2f3256(265)]();
  }
  [_0x596997(256)](_0x4e6f75) {
    var _0x2f74ac = _0x596997;
    this[_0x2f74ac(253)][_0x2f74ac(256)](_0x4e6f75);
  }
}
function _0x132a() {
  const _0x2a3a3b = ["need_phone_number", "uzTKh", "6057AivLtb", "uwnHn", "HGfhC", "CQqAz", "number", "photo_size", "DVHQp", "ZzQoz", 'Each price must have a "label" string', "rBVQQ", "gqzrv", "boolean", "suggested_tip_amounts", "HIQua", "HyvFb", "amount", "766LIWyYT", "title", "1924QpDzMU", "720654HMVynz", "WTJsZ", "string", "PsmCh", "GPgIk", "label", "CsKDX", "max_tip_amount", 'Field "', "description", 'Field "suggested_tip_amounts" must be an array if provided', "mtqCV", "zdvZF", "photo_width", "4095623HkfDVS", "cUONK", "ZUZDX", 'All values in "suggested_tip_amounts" must be numbers', "vIzzY", '" is required and must be a string', "length", "yfNer", "currency", "mhRHy", "lBLFi", "6156265yVvJNN", "is_flexible", "higjs", "TvmdN", "Payload is required", "HhVLk", "send_email_to_provider", "PVvOt", "fMkTV", "xglHG", 'Each price must have a positive "amount" number', "slug", "BtNSb", "payload", "need_name", "14661wxWYAo", "1956ImwpLs", "need_shipping_address", "subscription_period", "DIMIO", "photo_height", "ZapiC", "isArray", "19726420rChwKq", "flUNc", "2984QnfaWC", "XAlZY"];
  _0x132a = function() {
    return _0x2a3a3b;
  };
  return _0x132a();
}
(function(_0x38aec1, _0x34fb32) {
  const _0x1914b3 = _0x36e7, _0x477b99 = _0x38aec1();
  while (!![]) {
    try {
      const _0x1c3df6 = -parseInt(_0x1914b3(203)) / 1 * (parseInt(_0x1914b3(159)) / 2) + -parseInt(_0x1914b3(143)) / 3 * (-parseInt(_0x1914b3(161)) / 4) + -parseInt(_0x1914b3(187)) / 5 + -parseInt(_0x1914b3(162)) / 6 + -parseInt(_0x1914b3(176)) / 7 + -parseInt(_0x1914b3(139)) / 8 * (-parseInt(_0x1914b3(202)) / 9) + parseInt(_0x1914b3(137)) / 10;
      if (_0x1c3df6 === _0x34fb32) break;
      else _0x477b99["push"](_0x477b99["shift"]());
    } catch (_0x16b8bd) {
      _0x477b99["push"](_0x477b99["shift"]());
    }
  }
})(_0x132a, 865799);
function _0x36e7(_0x4f0a3e, _0x49d645) {
  const _0x132a87 = _0x132a();
  return _0x36e7 = function(_0x36e7e2, _0x3c5c9e) {
    _0x36e7e2 = _0x36e7e2 - 137;
    let _0x87a886 = _0x132a87[_0x36e7e2];
    return _0x87a886;
  }, _0x36e7(_0x4f0a3e, _0x49d645);
}
function validateInvoicePayload(_0x3cfb85) {
  const _0x16e85a = _0x36e7, _0x5e2bc6 = { "zdvZF": function(_0x496aa3, _0x5480d0) {
    return _0x496aa3 !== _0x5480d0;
  }, "yfNer": _0x16e85a(164), "DIMIO": function(_0x10851a, _0x488198) {
    return _0x10851a !== _0x488198;
  }, "flUNc": _0x16e85a(179), "BtNSb": _0x16e85a(151), "GPgIk": function(_0x3edc3d, _0x26a708) {
    return _0x3edc3d !== _0x26a708;
  }, "TvmdN": _0x16e85a(147), "PsmCh": function(_0x5b7c70, _0x4bcc3d) {
    return _0x5b7c70 <= _0x4bcc3d;
  }, "HIQua": 'Each price must have a positive "amount" number', "higjs": _0x16e85a(172), "BtrZR": _0x16e85a(191), "gqzrv": _0x16e85a(198), "mhRHy": _0x16e85a(160), "CsKDX": _0x16e85a(200), "mtqCV": _0x16e85a(184), "uzTKh": function(_0x20f155, _0x46ac62) {
    return _0x20f155 === _0x46ac62;
  }, "ZapiC": 'Field "prices" must be a non-empty array', "YZhiy": _0x16e85a(149), "yqfwb": _0x16e85a(169), "uwnHn": _0x16e85a(175), "HyvFb": _0x16e85a(207), "PVvOt": function(_0xb24a04, _0x21713c) {
    return _0xb24a04 in _0x21713c;
  }, "fMkTV": function(_0x3b0c5f, _0x4fc14e) {
    return _0x3b0c5f !== _0x4fc14e;
  }, "CQqAz": _0x16e85a(177), "lBLFi": _0x16e85a(155), "HhVLk": function(_0x2897a1, _0x6baac1) {
    return _0x2897a1 !== _0x6baac1;
  }, "ZzQoz": _0x16e85a(180), "ZUZDX": _0x16e85a(204), "HGfhC": "send_phone_number_to_provider", "xglHG": _0x16e85a(193), "rQiIj": function(_0x42fc3a, _0x8222e2) {
    return _0x42fc3a !== _0x8222e2;
  }, "XAlZY": _0x16e85a(152), "QEQYT": _0x16e85a(163) };
  if (!_0x3cfb85) throw new Error(_0x5e2bc6["BtrZR"]);
  const _0x798870 = [_0x5e2bc6[_0x16e85a(153)], _0x5e2bc6[_0x16e85a(185)], _0x16e85a(171), _0x5e2bc6[_0x16e85a(168)], _0x5e2bc6[_0x16e85a(173)]];
  for (const _0x5df802 of _0x798870) {
    if (!_0x3cfb85[_0x5df802] || typeof _0x3cfb85[_0x5df802] !== _0x16e85a(164)) {
      if (_0x5e2bc6[_0x16e85a(142)]("rJnOU", "jmHVS")) {
        if (!_0x37dd8d[_0x119592] || _0x5e2bc6["zdvZF"](typeof _0x4cf029[_0x1783e4], _0x5e2bc6[_0x16e85a(183)])) throw new _0x3e9b32(_0x16e85a(170) + _0x40556f + _0x16e85a(181));
      } else throw new Error(_0x16e85a(170) + _0x5df802 + _0x16e85a(181));
    }
  }
  if (!Array["isArray"](_0x3cfb85["prices"]) || _0x5e2bc6[_0x16e85a(142)](_0x3cfb85["prices"][_0x16e85a(182)], 0)) throw new Error(_0x5e2bc6[_0x16e85a(208)]);
  for (const _0x1cba7c of _0x3cfb85["prices"]) {
    if (!_0x1cba7c[_0x16e85a(167)] || _0x5e2bc6[_0x16e85a(166)](typeof _0x1cba7c[_0x16e85a(167)], _0x5e2bc6["yfNer"])) {
      if (_0x5e2bc6["YZhiy"] === _0x16e85a(149)) throw new Error(_0x5e2bc6[_0x16e85a(199)]);
      else throw new _0x4ba65c(_0x16e85a(197));
    }
    if (_0x5e2bc6[_0x16e85a(166)](typeof _0x1cba7c[_0x16e85a(158)], _0x5e2bc6["TvmdN"]) || _0x5e2bc6[_0x16e85a(165)](_0x1cba7c["amount"], 0)) throw new Error(_0x5e2bc6[_0x16e85a(156)]);
  }
  const _0x10c63f = [_0x16e85a(205), _0x5e2bc6["yqfwb"], _0x16e85a(148), _0x5e2bc6[_0x16e85a(144)], _0x5e2bc6[_0x16e85a(157)]];
  for (const _0x43d8f9 of _0x10c63f) {
    if (_0x5e2bc6[_0x16e85a(194)](_0x43d8f9, _0x3cfb85) && _0x5e2bc6[_0x16e85a(195)](typeof _0x3cfb85[_0x43d8f9], _0x16e85a(147))) {
      if (_0x5e2bc6[_0x16e85a(195)](_0x5e2bc6[_0x16e85a(146)], _0x5e2bc6[_0x16e85a(146)])) {
        if (_0x5e2bc6[_0x16e85a(206)](typeof _0x553850, _0x16e85a(147))) throw new _0x272506(_0x5e2bc6[_0x16e85a(138)]);
      } else throw new Error(_0x16e85a(170) + _0x43d8f9 + '" must be a number if provided');
    }
  }
  if (_0x5e2bc6[_0x16e85a(194)](_0x5e2bc6[_0x16e85a(186)], _0x3cfb85)) {
    if (!Array["isArray"](_0x3cfb85[_0x16e85a(155)])) {
      if (_0x5e2bc6[_0x16e85a(192)](_0x5e2bc6["ZzQoz"], _0x5e2bc6[_0x16e85a(150)])) {
        if (!_0x133e8d[_0x16e85a(167)] || _0x5e2bc6[_0x16e85a(174)](typeof _0x1c133f[_0x16e85a(167)], _0x5e2bc6[_0x16e85a(183)])) throw new _0x4cf304(_0x5e2bc6["BtNSb"]);
        if (_0x5e2bc6[_0x16e85a(166)](typeof _0x473c5c[_0x16e85a(158)], _0x5e2bc6[_0x16e85a(190)]) || _0x5e2bc6[_0x16e85a(165)](_0x2ffd31[_0x16e85a(158)], 0)) throw new _0x4ca79d(_0x5e2bc6[_0x16e85a(156)]);
      } else throw new Error(_0x5e2bc6["higjs"]);
    }
    for (const _0x26135e of _0x3cfb85[_0x16e85a(155)]) {
      if (typeof _0x26135e !== _0x5e2bc6[_0x16e85a(190)]) throw new Error(_0x5e2bc6[_0x16e85a(138)]);
    }
  }
  const _0x2c512b = [_0x16e85a(201), _0x16e85a(141), "need_email", _0x5e2bc6[_0x16e85a(178)], _0x5e2bc6[_0x16e85a(145)], _0x5e2bc6[_0x16e85a(196)], _0x16e85a(188)];
  for (const _0x4fb2a4 of _0x2c512b) {
    if (_0x4fb2a4 in _0x3cfb85 && _0x5e2bc6["HhVLk"](typeof _0x3cfb85[_0x4fb2a4], _0x16e85a(154))) {
      if (_0x5e2bc6["rQiIj"](_0x5e2bc6[_0x16e85a(140)], _0x5e2bc6["QEQYT"])) throw new Error(_0x16e85a(170) + _0x4fb2a4 + '" must be a boolean if provided');
      else {
        if (!_0x39c981[_0x16e85a(209)](_0x4ffdd1[_0x16e85a(155)])) throw new _0x251874(_0x5e2bc6[_0x16e85a(189)]);
        for (const _0x4101c7 of _0xa5b164[_0x16e85a(155)]) {
          if (typeof _0x4101c7 !== _0x5e2bc6[_0x16e85a(190)]) throw new _0x652997(_0x16e85a(179));
        }
      }
    }
  }
}
function _0x3b9a(_0x1e3732, _0x4b32b3) {
  const _0x30f304 = _0x30f3();
  return _0x3b9a = function(_0x3b9afe, _0x596006) {
    _0x3b9afe = _0x3b9afe - 111;
    let _0xabe15e = _0x30f304[_0x3b9afe];
    return _0xabe15e;
  }, _0x3b9a(_0x1e3732, _0x4b32b3);
}
const _0x45d476 = _0x3b9a;
(function(_0x541415, _0x493866) {
  const _0x5c2ec0 = _0x3b9a, _0x3f39c0 = _0x541415();
  while (!![]) {
    try {
      const _0x430ddf = -parseInt(_0x5c2ec0(119)) / 1 * (parseInt(_0x5c2ec0(111)) / 2) + -parseInt(_0x5c2ec0(115)) / 3 * (-parseInt(_0x5c2ec0(120)) / 4) + parseInt(_0x5c2ec0(113)) / 5 + parseInt(_0x5c2ec0(114)) / 6 + parseInt(_0x5c2ec0(122)) / 7 * (parseInt(_0x5c2ec0(118)) / 8) + -parseInt(_0x5c2ec0(121)) / 9 + -parseInt(_0x5c2ec0(112)) / 10 * (parseInt(_0x5c2ec0(116)) / 11);
      if (_0x430ddf === _0x493866) break;
      else _0x3f39c0["push"](_0x3f39c0["shift"]());
    } catch (_0x4df2ca) {
      _0x3f39c0["push"](_0x3f39c0["shift"]());
    }
  }
})(_0x30f3, 995596);
function _0x30f3() {
  const _0x636919 = ["init", "313942iyqWWm", "6170aSBzqC", "375580LrZgfe", "10170504jYYZsS", "2481FsMzLJ", "11561vYcgBI", "PROD", "2224544PzmVIU", "3hbNaRY", "596YXITrZ", "3011247RmfXrh", "14FmoJXf", "dmBhB"];
  _0x30f3 = function() {
    return _0x636919;
  };
  return _0x30f3();
}
let __registerInvoice;
async function init({ token: _0x3b447c, appName: _0x49ef97, env = _0x45d476(117) }) {
  const _0x4f845f = _0x45d476, _0x138f91 = { "dmBhB": function(_0x1f6d49, _0x274ce9) {
    return _0x1f6d49(_0x274ce9);
  } }, _0x471243 = new App(_0x3b447c, _0x49ef97, env);
  __registerInvoice = (_0x7146b7) => {
    const _0x458325 = _0x3b9a;
    _0x138f91[_0x458325(123)](validateInvoicePayload, _0x7146b7), _0x471243["registerInvoice"](_0x7146b7);
  }, await _0x471243[_0x4f845f(124)]();
}
const index = { "init": init, "registerInvoice": (_0x2d3010) => __registerInvoice(_0x2d3010) };
export {
  index as default
};
